--- a/CMakeLists.txt.orig	2022-09-26 02:39:25.000000000 +0800
+++ b/CMakeLists.txt	2022-11-18 00:44:34.271140903 +0800
@@ -499,10 +499,3 @@
 	PATTERN "*.broker" EXCLUDE
 	PATTERN "python/plug/python" EXCLUDE
 	PATTERN "*")
-
-# setup some symlinks and remove deprecated stuff from previous installation
-install(CODE "
-	execute_process(COMMAND ln -sf ../../bin/far2l ${CMAKE_INSTALL_PREFIX}/lib/far2l/far2l_askpass)
-	execute_process(COMMAND ln -sf ../../bin/far2l ${CMAKE_INSTALL_PREFIX}/lib/far2l/far2l_sudoapp)
-	execute_process(COMMAND rm -f ${CMAKE_INSTALL_PREFIX}/lib/far2l/Plugins/objinfo/plug/objinfo.far-plug-mb)
-	" COMPONENT system)
