diff -urN hwsetup-1.2.fastprobe/hwsetup.c hwsetup-1.2/hwsetup.c
--- hwsetup-1.2.fastprobe/hwsetup.c	2008-04-07 19:09:08.000000000 -0700
+++ hwsetup-1.2/hwsetup.c	2008-04-07 19:07:55.000000000 -0700
@@ -388,7 +388,7 @@
          sprintf(fullpath,"%.90s%.32s",xpath,server[i]);
          if(!exists(fullpath)) continue;
          strncpy(xi.xserver,server[i],sizeof(xi.xserver));
-	}
+        }
       }
     }
    if((xpos=strstr(d->driver,xorg))!=NULL) /* Check for Xorg */
@@ -653,7 +653,7 @@
 int writeconfig(char *name,struct device *d,int verbose)
 {
  FILE *f,*k;
- const char *kconfig="/etc/sysconfig/knoppix";
+ const char *kconfig="/etc/sysconfig/gentoo";
  char *desc;
  unlink(name);
  if((f=fopen(name,"w"))==NULL)
