--- a/src/github.com/ullaakut/cameradar/cmd/cameradar/cameradar.go.orig	2019-05-26 14:33:08.000000000 +0800
+++ b/src/github.com/ullaakut/cameradar/cmd/cameradar/cameradar.go	2019-05-27 21:32:12.006597530 +0800
@@ -20,8 +20,8 @@
 
 	pflag.StringSliceP("targets", "t", []string{}, "The targets on which to scan for open RTSP streams - required (ex: 172.16.100.0/24)")
 	pflag.StringSliceP("ports", "p", []string{"554", "5554", "8554"}, "The ports on which to search for RTSP streams")
-	pflag.StringP("custom-routes", "r", "<GOPATH>/src/github.com/ullaakut/cameradar/dictionaries/routes", "The path on which to load a custom routes dictionary")
-	pflag.StringP("custom-credentials", "c", "<GOPATH>/src/github.com/ullaakut/cameradar/dictionaries/credentials.json", "The path on which to load a custom credentials JSON dictionary")
+	pflag.StringP("custom-routes", "r", "/usr/share/cameradar/routes", "The path on which to load a custom routes dictionary")
+	pflag.StringP("custom-credentials", "c", "/usr/share/cameradar/credentials.json", "The path on which to load a custom credentials JSON dictionary")
 	pflag.IntP("speed", "s", 4, "The nmap speed preset to use for discovery")
 	pflag.DurationP("timeout", "T", 2*time.Second, "The timeout in miliseconds to use for attack attempts")
 	pflag.BoolP("debug", "d", true, "Enable the debug logs")
