diff -ur a/Makefile b/Makefile
--- a/Makefile	2009-10-24 14:01:57.000000000 +0400
+++ b/Makefile	2019-10-26 23:14:50.410300745 +0300
@@ -16,7 +16,7 @@
 LOGFILE = /var/log/$(APP).log
 
 CPPC = g++
-CPPARGS = -pipe -Wall `pkg-config --cflags --libs gtk+-2.0 gdk-2.0 gthread-2.0 libgnome-2.0 libgnomeui-2.0 libnotify` -lpcap -O2 -Xlinker -s
+CXXFLAGS += `pkg-config --cflags --libs gtk+-2.0 gdk-2.0 gthread-2.0 libgnome-2.0 libgnomeui-2.0 libnotify` -lpcap
 
 HDR = antispoof/arpdef.h antispoof/antispoof.h antispoof/config.h antispoof/platform.h antispoof/protector.h antispoof/arpreplier.h antispoof/reply4hosts.h antispoof/common.h linux-ui/dialogs.h linux-ui/configdlg.h linux-ui/uicommon.h
 
@@ -25,7 +25,7 @@
 all: $(HDR) $(SRC)
 	echo "#define APP \"$(APP)\"" > config.h;\
 	echo "#define PREFIX \"$(prefix)\"" >> config.h;\
-	$(CPPC) $(CPPARGS) $(SRC) -o $(APP);\
+	$(CPPC) $(CXXFLAGS) $(SRC) -o $(APP);\
 	echo "#%PAM-1.0" > $(TPAMD);\
 	echo "auth		include		config-util" >> $(TPAMD);\
 	echo "account		include		config-util" >> $(TPAMD);\
@@ -40,13 +40,6 @@
 	mkdir -p $(DESTDIR)$(datadir)/pixmaps/$(APP);\
 	cp pixmaps/*.svg $(DESTDIR)$(datadir)/pixmaps/$(APP);\
 	ln -s $(datadir)/pixmaps/$(APP)/icon.svg $(DESTDIR)$(datadir)/pixmaps/arpas.svg;\
-	mkdir -p $(DESTDIR)$(datadir)/applications;\
-	cp arpas.desktop $(DESTDIR)$(datadir)/applications/;\
-	mkdir -p $(DESTDIR)$(DOCDIR);\
-	cp ChangeLog $(DESTDIR)$(DOCDIR);\
-	cp readme.txt $(DESTDIR)$(DOCDIR)/README;\
-	mkdir -p $(DESTDIR)$(bindir);\
-	ln -s $(CONHELP) $(DESTDIR)$(bindir)/$(APP);\
 	mkdir -p $(DESTDIR)$(PAMD);\
 	cp $(TPAMD) $(DESTDIR)$(PAMD)/$(APP);\
 	mkdir -p $(DESTDIR)$(CONSAPP);\
