diff -r -u ../kismet-2007-10-R1/kismet_wrapper.cc kismet-2007-10-R1/kismet_wrapper.cc
--- ../kismet-2007-10-R1/kismet_wrapper.cc	2007-10-06 23:50:03.000000000 +0200
+++ kismet-2007-10-R1/kismet_wrapper.cc	2008-03-24 14:23:51.831523343 +0100
@@ -28,6 +28,8 @@
 
 #include "config.h"
 
+#include <cstring>
+
 /* Blob of globals since sighandler needs them */
 vector<string> postcli_err;
 pid_t srvpid = -1, clipid = -1;
diff -r -u ../kismet-2007-10-R1/ringbuf.cc kismet-2007-10-R1/ringbuf.cc
--- ../kismet-2007-10-R1/ringbuf.cc	2005-12-11 20:14:39.000000000 +0100
+++ kismet-2007-10-R1/ringbuf.cc	2008-03-24 14:23:51.875510223 +0100
@@ -18,6 +18,8 @@
 
 #include "ringbuf.h"
 
+#include <cstring>
+
 RingBuffer::RingBuffer(int in_size) {
     ring_len = in_size;
     ring_data = new uint8_t[in_size];
diff -r -u ../kismet-2007-10-R1/util.cc kismet-2007-10-R1/util.cc
--- ../kismet-2007-10-R1/util.cc	2006-08-28 22:37:29.000000000 +0200
+++ kismet-2007-10-R1/util.cc	2008-03-24 14:23:51.875510223 +0100
@@ -23,6 +23,8 @@
 #include <sys/stat.h>
 #include <fcntl.h>
 
+#include <cstring>
+
 // We need this to make uclibc happy since they don't even have rintf...
 #ifndef rintf
 #define rintf(x) (float) rint((double) (x))
