--- install.rdf.orig	2010-12-22 21:22:35.000000000 +0000
+++ install.rdf	2010-12-22 21:22:45.000000000 +0000
@@ -5,7 +5,7 @@
   <RDF:Description RDF:about="rdf:#$3yxCW3"
                    em:id="{ec8030f7-c20a-464f-9b0e-13a3a9e97384}"
                    em:minVersion="3.0"
-                   em:maxVersion="3.6a1pre" />
+                   em:maxVersion="3.6.*" />
   <RDF:Description RDF:about="urn:mozilla:install-manifest"
                    em:id="{E2082660-5330-49e6-BD84-9978CE15BA72}"
                    em:name="SQL Injection!"
