diff --git a/ruby/Makefile.am b/ruby/Makefile.am
--- a/ruby/Makefile.am
+++ b/ruby/Makefile.am
@@ -44,11 +44,11 @@
 	RUBY=$(RUBY) RAKE=$(RAKE)
 
 all:
-	$(RAKE) build
-	$(RAKE) rdoc
+	$(RAKE) -v build
+	$(RAKE) -v rdoc
 
-RUBY_SITELIB := $(shell $(RUBY) -rrbconfig -e "puts Config::CONFIG['sitelibdir']")
-RUBY_SITEARCH := $(shell $(RUBY) -rrbconfig -e "puts Config::CONFIG['sitearchdir']")
+RUBY_SITELIB := $(shell $(RUBY) -rrbconfig -e "puts RbConfig::CONFIG['sitelibdir']")
+RUBY_SITEARCH := $(shell $(RUBY) -rrbconfig -e "puts RbConfig::CONFIG['sitearchdir']")
 
 install:
 	$(MKDIR_P) $(DESTDIR)$(RUBY_SITELIB)
diff --git a/ruby/Rakefile.in b/ruby/Rakefile.in
--- a/ruby/Rakefile.in
+++ b/ruby/Rakefile.in
@@ -17,9 +17,9 @@
 # Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 
 require 'rake/clean'
-require 'rake/rdoctask'
+require 'rdoc/task'
 require 'rake/testtask'
-require 'rake/gempackagetask'
+require 'rubygems/package_task'
 
 PKG_NAME='@PACKAGE_NAME@'
 PKG_VERSION='@PACKAGE_VERSION@'
@@ -108,7 +108,7 @@
 EOF
 end
 
-Rake::GemPackageTask.new(SPEC) do |pkg|
+Gem::PackageTask.new(SPEC) do |pkg|
     pkg.need_tar = true
     pkg.need_zip = true
 end
