--- common/lib/modules/fglrx/build_mod/kcl_debug.h.orig	2012-03-08 19:30:38.195025328 +0100
+++ common/lib/modules/fglrx/build_mod/kcl_debug.h	2012-03-08 19:31:24.976024507 +0100
@@ -85,8 +85,8 @@
 #ifdef ERROR
 #undef ERROR
 #endif
-#ifdef WARN
-#undef WARN
+#ifdef AMD_WARN
+#undef AMD_WARN
 #endif
 #ifdef INFO
 #undef INFO
@@ -122,7 +122,7 @@
 {
     SPECIAL = 0,
     ERROR  ,
-    WARN  ,
+    AMD_WARN  ,
     INFO ,
     INFOEX,
     TRACE,
@@ -160,7 +160,7 @@
 }log_map;
 
 
-#define DEFAULT_LOG_LEVEL ((U08)(___BIT(INFO) | ___BIT(INFOEX) |___BIT(ERROR) |___BIT(WARN) | ___BIT(TRACE)| ___BIT(SPECIAL)  ))
+#define DEFAULT_LOG_LEVEL ((U08)(___BIT(INFO) | ___BIT(INFOEX) |___BIT(ERROR) |___BIT(AMD_WARN) | ___BIT(TRACE)| ___BIT(SPECIAL)  ))
 #define INFO_LOG_LEVEL ((U08)(___BIT(INFO) | ___BIT(INFOEX)))
 extern const log_map module_log_map[];
 extern const module_map module_type_map[];
--- common/lib/modules/fglrx/build_mod/kcl_debug.c.orig	2012-03-08 19:36:18.244019310 +0100
+++ common/lib/modules/fglrx/build_mod/kcl_debug.c	2012-03-08 19:36:36.197018973 +0100
@@ -69,7 +69,7 @@
 {
     {SPECIAL        ,   'S'},
     {ERROR          ,   'E'},
-    {WARN           ,   'W'},
+    {AMD_WARN           ,   'W'},
     {INFO           ,   'I'},
     {INFOEX         ,   'X'},
     {TRACE          ,   'T'},
