--- pentoo-installer-orig	2013-07-02 23:44:02.497581000 -0400
+++ pentoo-installer	2013-07-03 00:27:02.535873000 -0400
@@ -708,9 +708,11 @@
 
     rsync -av /mnt/livecd/* ${DESTDIR} 2>&1 >$LOG
     rsync -av /etc/* ${DESTDIR}/etc/ 2>&1 >>$LOG
+    mv ${DESTDIR}/etc/inittab.old ${DISTDIR}/etc/inittab 2>&1 >>$LOG
     rsync -av /root/* ${DESTDIR}/root/ 2>&1 >>$LOG
     rsync -av /usr/portage ${DESTDIR}/usr/ >>$LOG
     rsync -av /var/lib/layman/pentoo ${DESTDIR}/var/lib/layman/ >>$LOG
+    ROOT=${DISTDIR} emerge -C app-misc/livecd-tools pentoo/pentoo-livecd
     if [ $? -ne 0 ]; then
         DIALOG --msgbox "Rsync failed (maybe you're out of disk space?). See the log output for more information"
         return 1
