--- setup.orig	2011-03-12 04:49:38.000000000 +0800
+++ setup	2013-03-07 15:22:02.724019430 +0800
@@ -755,10 +755,10 @@
     cp -a $DESTDIR/usr/lib/grub/i386-pc/* $DESTDIR/boot/grub/
     sync
     # freeze xfs filesystems to enable grub installation on xfs filesystems
-    if [ -x /usr/sbin/xfs_freeze ]; then
-        /usr/sbin/xfs_freeze -f $DESTDIR/boot > /dev/null 2>&1
-        /usr/sbin/xfs_freeze -f $DESTDIR/ > /dev/null 2>&1
-    fi
+#    if [ -x /usr/sbin/xfs_freeze ]; then
+#        /usr/sbin/xfs_freeze -f $DESTDIR/boot > /dev/null 2>&1
+#        /usr/sbin/xfs_freeze -f $DESTDIR/ > /dev/null 2>&1
+#    fi
     # look for a separately-mounted /boot partition
     bootpart=$(mount | grep $DESTDIR/boot | cut -d' ' -f 1)
     if [ "$bootpart" = "" ]; then
@@ -787,10 +787,10 @@
     /sbin/grub-install --no-floppy --recheck --grub-shell=$DESTDIR/sbin/grub --root-directory=$DESTDIR $ROOTDEV >/tmp/grub.log 2>&1
     cat /tmp/grub.log >$LOG
     # unfreeze xfs filesystems
-    if [ -x /usr/sbin/xfs_freeze ]; then
-        /usr/sbin/xfs_freeze -u $DESTDIR/boot > /dev/null 2>&1
-        /usr/sbin/xfs_freeze -u $DESTDIR/ > /dev/null 2>&1
-    fi
+#    if [ -x /usr/sbin/xfs_freeze ]; then
+#        /usr/sbin/xfs_freeze -u $DESTDIR/boot > /dev/null 2>&1
+#        /usr/sbin/xfs_freeze -u $DESTDIR/ > /dev/null 2>&1
+#    fi
 
     if grep "Error [0-9]*: " /tmp/grub.log >/dev/null; then
         DIALOG --msgbox "Error installing GRUB. (see $LOG for output)" 0 0
