--- setup	2013-03-06 12:13:09.978978369 -0500
+++ setup	2013-03-06 12:11:47.490976160 -0500
@@ -1148,6 +1148,9 @@
 				chroot ${DESTDIR} useradd -m -G users,wheel,audio,cdrom,video,cdrw,plugdev $_username
 				rsync -r --exclude=.svn --exclude=.subversion "${DESTDIR}"/root/.[!.]* "${DESTDIR}"/home/"${_username}"/
 				chroot ${DESTDIR} chown -R ${_username}:${_username} /home/$_username
+			while true; do
+				chroot ${DESTDIR} passwd $_username && break
+			done
         else                                                #regular file
             $EDITOR ${DESTDIR}${CURRENT_SELECTION}
         fi
