diff -Naur user-agent-switcher-0.6.11-fx+sm.orig/install.rdf user-agent-switcher-0.6.11-fx+sm/install.rdf
--- user-agent-switcher-0.6.11-fx+sm.orig/install.rdf   2008-03-17 17:37:14.000000000 +0000
+++ user-agent-switcher-0.6.11-fx+sm/install.rdf        2009-01-27 12:13:01.000000000 +0000
@@ -24,7 +24,7 @@
         <em:targetApplication>
             <Description>
                 <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
-                <em:maxVersion>3.0b4</em:maxVersion>
+                <em:maxVersion>3.1</em:maxVersion>
                 <em:minVersion>1.0</em:minVersion>
             </Description>
         </em:targetApplication>
@@ -43,4 +43,4 @@
             </Description>
         </em:targetApplication>
     </Description>
-</RDF>
\ No newline at end of file
+</RDF>

