--- ./drivers/gpu/drm/nouveau/nouveau_fence.h
+++ ./drivers/gpu/drm/nouveau/nouveau_fence.h
@@ -43,7 +43,7 @@
 	int  (*sync)(struct nouveau_fence *, struct nouveau_channel *,
 		     struct nouveau_channel *);
 	u32  (*read)(struct nouveau_channel *);
-};
+} __no_const;
 
 #define nouveau_fence(drm) ((struct nouveau_fence_priv *)(drm)->fence)
 
--- ./drivers/gpu/drm/nouveau/nouveau_bios.c
+++ ./drivers/gpu/drm/nouveau/nouveau_bios.c
@@ -1015,7 +1015,7 @@
 struct bit_table {
 	const char id;
 	int (* const parse_fn)(struct drm_device *, struct nvbios *, struct bit_entry *);
-};
+} __no_const;
 
 #define BIT_TABLE(id, funcid) ((struct bit_table){ id, parse_bit_##funcid##_tbl_entry })
 
--- ./include/drm/drm_crtc_helper.h
+++ ./include/drm/drm_crtc_helper.h
@@ -109,7 +109,7 @@
 					    struct drm_connector *connector);
 	/* disable encoder when not in use - more explicit than dpms off */
 	void (*disable)(struct drm_encoder *encoder);
-};
+} __no_const;
 
 /**
  * drm_connector_helper_funcs - helper operations for connectors
