diff --git a/vmblock-only/Makefile b/vmblock-only/Makefile
index 0a9ffd6..4aa09e8 100644
--- a/vmblock-only/Makefile
+++ b/vmblock-only/Makefile
@@ -29,9 +29,9 @@ SRCROOT = .
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = $(KERNEL_DIR) 
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = $(KBUILD_OUTPUT)
 
 DRIVER := vmblock
 PRODUCT := @PRODUCT@
diff --git a/vmci-only/Makefile b/vmci-only/Makefile
index 0a6c9c6..6cd0b6d 100644
--- a/vmci-only/Makefile
+++ b/vmci-only/Makefile
@@ -29,9 +29,9 @@ SRCROOT = .
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = $(KERNEL_DIR)
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = $(KBUILD_OUTPUT)
 
 DRIVER := vmci
 PRODUCT := @PRODUCT@
diff --git a/vmmon-only/Makefile b/vmmon-only/Makefile
index d26e58c..f8a40f4 100644
--- a/vmmon-only/Makefile
+++ b/vmmon-only/Makefile
@@ -29,9 +29,9 @@ SRCROOT = .
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = $(KERNEL_DIR)
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = $(KBUILD_OUTPUT)
 
 DRIVER := vmmon
 PRODUCT := @PRODUCT@
diff --git a/vmnet-only/Makefile b/vmnet-only/Makefile
index 4cabc30..568271a 100644
--- a/vmnet-only/Makefile
+++ b/vmnet-only/Makefile
@@ -29,9 +29,9 @@ SRCROOT = .
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = $(KERNEL_DIR)
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = $(KBUILD_OUTPUT)
 
 DRIVER := vmnet
 PRODUCT := @PRODUCT@
diff --git a/vsock-only/Makefile b/vsock-only/Makefile
index a69bc22..8a74856 100644
--- a/vsock-only/Makefile
+++ b/vsock-only/Makefile
@@ -29,9 +29,9 @@ SRCROOT = .
 VM_UNAME = $(shell uname -r)
 
 # Header directory for the running kernel
-HEADER_DIR = /lib/modules/$(VM_UNAME)/build/include
+HEADER_DIR = $(KERNEL_DIR)
 
-BUILD_DIR = $(HEADER_DIR)/..
+BUILD_DIR = $(KBUILD_OUTPUT)
 
 DRIVER := vsock
 PRODUCT := ws
