diff -Nurp work.orig/vmblock-only/linux/inode.c work/vmblock-only/linux/inode.c
--- work.orig/vmblock-only/linux/inode.c	2013-10-18 17:56:11.000000000 +0000
+++ work/vmblock-only/linux/inode.c	2013-11-14 16:44:41.079398115 +0000
@@ -135,7 +135,8 @@ InodeOpLookup(struct inode *dir,      //
    inode->i_size = INODE_TO_IINFO(inode)->nameLen;
    inode->i_version = 1;
    inode->i_atime = inode->i_mtime = inode->i_ctime = CURRENT_TIME;
-   inode->i_uid = inode->i_gid = 0;
+   inode->i_uid = KUIDT_INIT(0);
+   inode->i_gid = KGIDT_INIT(0);
    inode->i_op = &LinkInodeOps;
 
    d_add(dentry, inode);

