diff -rupN vmci-only/linux/driver.c vmci-only/linux/driver.c
--- vmci-only/linux/driver.c	2014-03-31 18:49:14.840009422 +0100
+++ vmci-only/linux/driver.c	2014-03-31 18:49:21.385009439 +0100
@@ -737,7 +737,7 @@ LinuxDriver_Ioctl(struct inode *inode,
          goto init_release;
       }
 
-      user = current_uid();
+      user = from_kuid(current_user_ns(), current_uid());
       retval = VMCIContext_InitContext(initBlock.cid, initBlock.flags,
                                        0 /* Unused */, vmciLinux->userVersion,
                                        &user, &vmciLinux->context);
diff -rupN vmci-only/shared/vm_assert.h vmci-only/shared/vm_assert.h
--- vmci-only/shared/vm_assert.h	2014-03-31 18:49:14.841009422 +0100
+++ vmci-only/shared/vm_assert.h	2014-03-31 18:49:21.385009439 +0100
@@ -256,6 +256,7 @@ void WarningThrottled(uint32 *count, con
                                        __FILE__, __LINE__, __FUNCTION__, \
                                        _fix))
 #else
+   #undef DEPRECATED /* in <linux/printk.h> since 3.14.0 */
    #define DEPRECATED(_fix) do {} while (0)
 #endif
 
diff -rupN vmmon-only/include/vm_assert.h vmmon-only/include/vm_assert.h
--- vmmon-only/include/vm_assert.h	2014-03-31 18:49:14.835009422 +0100
+++ vmmon-only/include/vm_assert.h	2014-03-31 18:49:21.380009439 +0100
@@ -256,6 +256,7 @@ void WarningThrottled(uint32 *count, con
                                        __FILE__, __LINE__, __FUNCTION__, \
                                        _fix))
 #else
+   #undef DEPRECATED /* in <linux/printk.h> since 3.14.0 */
    #define DEPRECATED(_fix) do {} while (0)
 #endif
 
diff -rupN vmmon-only/linux/driver.c vmmon-only/linux/driver.c
--- vmmon-only/linux/driver.c	2014-03-31 18:49:14.834009422 +0100
+++ vmmon-only/linux/driver.c	2014-03-31 18:49:21.379009439 +0100
@@ -1338,7 +1338,9 @@ LinuxDriverReadTSC(void *data,   // OUT:
  *-----------------------------------------------------------------------------
  */
 
-__attribute__((always_inline)) static Bool
+#include <linux/compiler-gcc.h>
+
+__always_inline static Bool
 LinuxDriverSyncReadTSCs(uint64 *delta) // OUT: TSC max - TSC min
 {
    TSCDelta tscDelta;
diff -rupN vmnet-only/vm_assert.h vmnet-only/vm_assert.h
--- vmnet-only/vm_assert.h	2014-03-31 18:49:14.845009422 +0100
+++ vmnet-only/vm_assert.h	2014-03-31 18:49:21.390009439 +0100
@@ -256,6 +256,7 @@ void WarningThrottled(uint32 *count, con
                                        __FILE__, __LINE__, __FUNCTION__, \
                                        _fix))
 #else
+   #undef DEPRECATED /* in <linux/printk.h> since 3.14.0 */
    #define DEPRECATED(_fix) do {} while (0)
 #endif
 
diff -rupN vsock-only/linux/af_vsock.c vsock-only/linux/af_vsock.c
--- vsock-only/linux/af_vsock.c	2014-03-31 18:49:14.837009422 +0100
+++ vsock-only/linux/af_vsock.c	2014-03-31 18:49:21.381009439 +0100
@@ -2869,7 +2869,7 @@ __VSockVmciCreate(struct net *net,
       vsk->connectTimeout = psk->connectTimeout;
    } else {
       vsk->trusted = capable(CAP_NET_ADMIN);
-      vsk->owner = current_uid();
+      vsk->owner = from_kuid(current_user_ns(), current_uid());
       vsk->queuePairSize = VSOCK_DEFAULT_QP_SIZE;
       vsk->queuePairMinSize = VSOCK_DEFAULT_QP_SIZE_MIN;
       vsk->queuePairMaxSize = VSOCK_DEFAULT_QP_SIZE_MAX;
diff -rupN vsock-only/shared/vm_assert.h vsock-only/shared/vm_assert.h
--- vsock-only/shared/vm_assert.h	2014-03-31 18:49:14.839009422 +0100
+++ vsock-only/shared/vm_assert.h	2014-03-31 18:49:21.382009439 +0100
@@ -256,6 +256,7 @@ void WarningThrottled(uint32 *count, con
                                        __FILE__, __LINE__, __FUNCTION__, \
                                        _fix))
 #else
+   #undef DEPRECATED /* in <linux/printk.h> since 3.14.0 */
    #define DEPRECATED(_fix) do {} while (0)
 #endif
 
