--- vmci-only/linux/driver.c.orig	2014-04-15 01:41:40.000000000 +0400
+++ vmci-only/linux/driver.c	2014-08-06 14:57:29.000000000 +0400
@@ -737,7 +737,7 @@
          goto init_release;
       }
 
-      user = current_uid();
+      user = current_uid().val;
       retval = VMCIContext_InitContext(initBlock.cid, initBlock.flags,
                                        0 /* Unused */, vmciLinux->userVersion,
                                        &user, &vmciLinux->context);
