diff -rupN vsock-only/linux/af_vsock.c vsock-only-mod/linux/af_vsock.c
--- vsock-only/linux/af_vsock.c	2014-04-15 03:11:41.000000000 +0530
+++ vsock-only-mod/linux/af_vsock.c	2014-05-27 21:39:51.081284121 +0530
@@ -2869,7 +2869,8 @@ __VSockVmciCreate(struct net *net,
       vsk->connectTimeout = psk->connectTimeout;
    } else {
       vsk->trusted = capable(CAP_NET_ADMIN);
-      vsk->owner = current_uid();
+	  kuid_t kern_uid = current_uid();
+      vsk->owner = kern_uid.val;
       vsk->queuePairSize = VSOCK_DEFAULT_QP_SIZE;
       vsk->queuePairMinSize = VSOCK_DEFAULT_QP_SIZE_MIN;
       vsk->queuePairMaxSize = VSOCK_DEFAULT_QP_SIZE_MAX;
diff -rupN vsock-only/linux/notify.c vsock-only-mod/linux/notify.c
--- vsock-only/linux/notify.c	2014-04-15 03:11:41.000000000 +0530
+++ vsock-only-mod/linux/notify.c	2014-05-27 21:36:56.713291909 +0530
@@ -516,7 +516,7 @@ VSockVmciHandleWrote(struct sock *sk,
    PKT_FIELD(vsk, sentWaitingRead) = FALSE;
 #endif
 
-   sk->sk_data_ready(sk, 0);
+   sk->sk_data_ready(sk);
 }
 
 
diff -rupN vsock-only/linux/notifyQState.c vsock-only-mod/linux/notifyQState.c
--- vsock-only/linux/notifyQState.c	2014-04-15 03:11:41.000000000 +0530
+++ vsock-only-mod/linux/notifyQState.c	2014-05-27 21:38:52.561286734 +0530
@@ -164,7 +164,7 @@ VSockVmciHandleWrote(struct sock *sk,
                      struct sockaddr_vm *dst,    // IN: unused
                      struct sockaddr_vm *src)    // IN: unused
 {
-   sk->sk_data_ready(sk, 0);
+   sk->sk_data_ready(sk);
 }
 
 
--- vsock-only/linux/notifyQState.c.orig	2014-04-15 01:41:41.000000000 +0400
+++ vsock-only/linux/notifyQState.c	2014-08-06 15:08:43.000000000 +0400
@@ -566,7 +566,7 @@
       }
 
       /* See the comment in VSockVmciNotifyPktSendPostEnqueue */
-      sk->sk_data_ready(sk, 0);
+      sk->sk_data_ready(sk);
    }
 
    return err;
