--- Core/loaders/checker/check_depen.py.orig	2016-07-17 07:01:33.000000000 +0800
+++ Core/loaders/checker/check_depen.py	2016-07-19 15:32:48.233392703 +0800
@@ -24,9 +24,5 @@
         if m[k] == False:
             if k == 0:notinstall('isc-dhcp-server')
             if k == 1:notinstall('hostapd')
-    if not path.isfile('Templates/Update/Windows_Update/Settins_WinUpdate.html'):
-        if '/usr/bin/wifi-pumpkin' in sys.argv[0]:
-            chdir('/usr/share/WiFi-Pumpkin')
-        copy('Settings/source.tar.gz','Templates/')
-        system('cd Templates/ && tar -xf source.tar.gz')
-        remove('Templates/source.tar.gz')
+    if '/usr/sbin/wifi-pumpkin' in sys.argv[0]:
+        chdir('/usr/share/wifi-pumpkin')
