--- a/setup.py	2020-06-02 21:35:06.878788116 +0200
+++ b/setup.py	2020-06-02 21:35:48.923792232 +0200
@@ -33,7 +33,7 @@
     author_email='team@datafolklabs.com',
     url='http://builtoncement.com',
     license='BSD',
-    packages=find_packages(exclude=['ez_setup', 'tests*']),
+    packages=find_packages(exclude=['ez_setup', 'tests*', 'examples']),
     include_package_data=True,
     zip_safe=False,
     test_suite='nose.collector',
