--- conf/neo4j-wrapper.conf.orig	2012-04-24 09:08:14.000000000 -0300
+++ conf/neo4j-wrapper.conf	2012-04-26 15:00:29.000000000 -0300
@@ -2,12 +2,12 @@
 # JVM Parameters
 #********************************************************************
  
-wrapper.java.additional.1=-Dorg.neo4j.server.properties=conf/neo4j-server.properties
-wrapper.java.additional.2=-Djava.util.logging.config.file=conf/logging.properties
+wrapper.java.additional.1=-Dorg.neo4j.server.properties=/etc/neo4j/neo4j-server.properties
+wrapper.java.additional.2=-Djava.util.logging.config.file=/etc/neo4j/logging.properties
 wrapper.java.additional.3=-Dfile.encoding=UTF-8
 
 # Uncomment the following line to enable garbage collection logging
-#wrapper.java.additional.4=-Xloggc:data/log/neo4j-gc.log
+#wrapper.java.additional.4=-Xloggc:/var/log/neo4j-gc.log
 
 # Initial Java Heap Size (in MB)
 #wrapper.java.initmemory=16
@@ -19,8 +19,8 @@
 # Wrapper settings
 #********************************************************************
 # Override default pidfile and lockfile 
-#wrapper.pidfile=../data/neo4j-server.pid
-#wrapper.lockfile=../data/neo4j-server.lck
+wrapper.pidfile=/var/db/neo4j/neo4j-server.pid
+wrapper.lockfile=/var/db/neo4j/neo4j-server.lck
 
 #********************************************************************
 # Wrapper Windows NT/2000/XP Service Properties
