diff -Naru o/Songbird/songbird n/Songbird/songbird
--- Songbird/songbird	2009-03-09 23:03:30.000000000 +0100
+++ Songbird/songbird	2009-03-17 11:05:34.000000000 +0100
@@ -324,11 +324,11 @@
 			break
 		fi
 		curdir=`dirname "$progname"`
-		run_moz="$curdir/$MOZ_DEFAULT_NAME"
+		run_moz="$curdir/songbird-bin"
 		if [ -x "$run_moz" ]; then
 			cd "$curdir"
 			MOZ_DIST_BIN=`pwd`
-			MOZ_PROGRAM=$MOZ_DIST_BIN/$MOZ_DEFAULT_NAME
+			MOZ_PROGRAM=$MOZ_DIST_BIN/songbird-bin
 			break
 		fi
 	done
