--- a/ui/setup.py
+++ b/ui/setup.py
@@ -28,11 +28,9 @@ These rules can last forever, until the app restart or just one time.',
       packages=find_packages(),
       include_package_data = True,
       package_data={'': ['*.*']},
-      data_files=[('/usr/share/applications', ['resources/opensnitch_ui.desktop']),
-               ('/usr/share/kservices5', ['resources/kcm_opensnitch.desktop']),
-               ('/usr/share/icons/hicolor/scalable/apps', ['resources/icons/opensnitch-ui.svg']),
-               ('/usr/share/icons/hicolor/48x48/apps', ['resources/icons/48x48/opensnitch-ui.png']),
-               ('/usr/share/icons/hicolor/64x64/apps', ['resources/icons/64x64/opensnitch-ui.png']),
-               ('/usr/share/metainfo', ['resources/io.github.evilsocket.opensnitch.appdata.xml'])],
+      data_files=[('share/applications', ['resources/opensnitch_ui.desktop']),
+               ('share/kservices5', ['resources/kcm_opensnitch.desktop']),
+               ('share/icons/hicolor/scalable/apps', ['resources/icons/opensnitch-ui.svg']),
+               ('share/metainfo', ['resources/io.github.evilsocket.opensnitch.appdata.xml'])],
       scripts = [ 'bin/opensnitch-ui' ],
       zip_safe=False)

