--- install.rdf.orig	2009-04-13 11:36:02.000000000 +0000
+++ install.rdf	2009-11-26 17:50:14.000000000 +0000
@@ -19,7 +19,7 @@
 			<Description>
 				<em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
 				<em:minVersion>1.5</em:minVersion>
-				<em:maxVersion>3.1b3</em:maxVersion>
+				<em:maxVersion>3.5.*</em:maxVersion>
 			</Description>
 		</em:targetApplication>
 
@@ -34,4 +34,4 @@
 
 	</Description>
 
-</RDF>
\ No newline at end of file
+</RDF>
