diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2014-09-16 23:08:06.000000000 +0400
+++ b/configure.ac	2019-03-17 10:08:12.594871130 +0300
@@ -150,7 +150,7 @@
 AC_ARG_ENABLE([flexscanners],
               AS_HELP_STRING([--disable-flexscanners], [disable FLEX-based scanners]),
               [],
-              [AC_DEFINE(FLEXSCANNERS_ENABLED, 1, [Use FLEX-based scanners]), flexscanners='yes'])
+              [AC_DEFINE(FLEXSCANNERS_ENABLED, 1, [Use FLEX-based scanners]) flexscanners='yes'])
 AM_CONDITIONAL([FLEXSCANNERS_ENABLED], [test "yes" = "$flexscanners"])
 
 
diff -ur a/m4/ax_boost_base.m4 b/m4/ax_boost_base.m4
--- a/m4/ax_boost_base.m4	2014-09-16 22:34:00.000000000 +0400
+++ b/m4/ax_boost_base.m4	2019-03-17 10:12:31.849532373 +0300
@@ -107,7 +107,7 @@
     dnl this location ist chosen if boost libraries are installed with the --layout=system option
     dnl or if you install boost with RPM
     if test "$ac_boost_path" != ""; then
-        BOOST_CPPFLAGS="-isystem$ac_boost_path/include"
+        BOOST_CPPFLAGS="-I$ac_boost_path/include"
         for ac_boost_path_tmp in $libsubdirs; do
                 if test -d "$ac_boost_path"/"$ac_boost_path_tmp" ; then
                         BOOST_LDFLAGS="-L$ac_boost_path/$ac_boost_path_tmp"
@@ -126,7 +126,7 @@
                     if ls "$ac_boost_path_tmp/$libsubdir/libboost_"* >/dev/null 2>&1 ; then break; fi
                 done
                 BOOST_LDFLAGS="-L$ac_boost_path_tmp/$libsubdir"
-                BOOST_CPPFLAGS="-isystem$ac_boost_path_tmp/include"
+                BOOST_CPPFLAGS="-I$ac_boost_path_tmp/include"
                 break;
             fi
         done
@@ -179,7 +179,7 @@
                         _version=$_version_tmp
                     fi
                     VERSION_UNDERSCORE=`echo $_version | sed 's/\./_/'`
-                    BOOST_CPPFLAGS="-isystem$ac_boost_path/include/boost-$VERSION_UNDERSCORE"
+                    BOOST_CPPFLAGS="-I$ac_boost_path/include/boost-$VERSION_UNDERSCORE"
                 done
             fi
         else
@@ -202,7 +202,7 @@
             done
 
             VERSION_UNDERSCORE=`echo $_version | sed 's/\./_/'`
-            BOOST_CPPFLAGS="-isystem$best_path/include/boost-$VERSION_UNDERSCORE"
+            BOOST_CPPFLAGS="-I$best_path/include/boost-$VERSION_UNDERSCORE"
             if test "$ac_boost_lib_path" = ""; then
                 for libsubdir in $libsubdirs ; do
                     if ls "$best_path/$libsubdir/libboost_"* >/dev/null 2>&1 ; then break; fi
@@ -221,7 +221,7 @@
                     V_CHECK=`expr $stage_version_shorten \>\= $_version`
                     if test "$V_CHECK" = "1" -a "$ac_boost_lib_path" = "" ; then
                         AC_MSG_NOTICE(We will use a staged boost library from $BOOST_ROOT)
-                        BOOST_CPPFLAGS="-isystem$BOOST_ROOT"
+                        BOOST_CPPFLAGS="-I$BOOST_ROOT"
                         BOOST_LDFLAGS="-L$BOOST_ROOT/stage/$libsubdir"
                     fi
                 fi
