--- src/node.cc.orig	2015-07-10 06:41:19.000000000 +0800
+++ src/node.cc	2015-07-31 17:33:25.035074443 +0800
@@ -2934,8 +2934,10 @@
          "                       present.\n"
 #endif
 #endif
+#if HAVE_OPENSSL
          "  --enable-ssl2        enable ssl2\n"
          "  --enable-ssl3        enable ssl3\n"
+#endif // HAVE_OPENSSL
          "\n"
          "Environment variables:\n"
 #ifdef _WIN32
@@ -3003,10 +3005,12 @@
     } else if (strcmp(arg, "--version") == 0 || strcmp(arg, "-v") == 0) {
       printf("%s\n", NODE_VERSION);
       exit(0);
+#if HAVE_OPENSSL
     } else if (strcmp(arg, "--enable-ssl2") == 0) {
       SSL2_ENABLE = true;
     } else if (strcmp(arg, "--enable-ssl3") == 0) {
       SSL3_ENABLE = true;
+#endif // HAVE_OPENSSL
     } else if (strcmp(arg, "--help") == 0 || strcmp(arg, "-h") == 0) {
       PrintHelp();
       exit(0);
