commit a07077f4737754757e99c2d42267f432289ffd13
Author: Magnus Holm <judofyr@gmail.com>
Date:   Fri Jun 24 10:43:55 2011 +0200

    Fix compilesite test for multiple threads (closes #93)

diff --git a/test/tilt_compilesite_test.rb b/test/tilt_compilesite_test.rb
index 1ba6433..2944c84 100644
--- a/test/tilt_compilesite_test.rb
+++ b/test/tilt_compilesite_test.rb
@@ -39,7 +39,7 @@ class CompileSiteTest < Test::Unit::TestCase
               locals = { "local#{i}" => 'value' }
               res = template.render(self, locals)
               thread_id = Thread.current.object_id
-              res = template.render(self, "local#{thread_id.to_s}" => 'value')
+              res = template.render(self, "local#{thread_id.abs.to_s}" => 'value')
             rescue => boom
               main_thread.raise(boom)
             end
