https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=5872

--- epan/wspython/Makefile.am	2011-04-29 06:57:10 +0000
+++ epan/wspython/Makefile.am	2011-04-29 07:15:20 +0000
@@ -44,12 +44,14 @@
 libwspython_la_CFLAGS = @PY_CFLAGS@
 
 wspythondir=@pythondir@
+wspythondissectorsdir=@pythondir@/wspy_dissectors
 
 wspython_DATA= \
   register-dissector.py \
   wspy_dissector.py \
   wspy_libws.py
 
+wspythondissectors_DATA=
 
 EXTRA_DIST = \
 	Makefile.common		\

