https://bugs.wireshark.org/bugzilla/show_bug.cgi?id=5800

Index: epan/dissectors/packet-ssl-utils.c
===================================================================
--- epan/dissectors/packet-ssl-utils.c	(revision 37067)
+++ epan/dissectors/packet-ssl-utils.c	(working copy)
@@ -946,7 +946,7 @@
     str->data_len = len;
 }
 
-#ifdef HAVE_LIBGNUTLS
+#if defined(HAVE_LIBGNUTLS) && defined(HAVE_LIBGCRYPT)
 
 static gint ver_major, ver_minor, ver_patch;
 
@@ -2747,7 +2747,7 @@
     sscanf(str, "%d.%d.%d", &ver_major, &ver_minor, &ver_patch);
 }
 
-#else /* HAVE_LIBGNUTLS */
+#else /* defined(HAVE_LIBGNUTLS) && defined(HAVE_LIBGCRYPT) */
 /* no libgnutl: dummy operation to keep interface consistent*/
 void
 ssl_lib_init(void)
