--- a/gtk/main.c
+++ b/gtk/main.c
@@ -2834,7 +2834,9 @@
 
   color_filters_init();
   decode_as_init();
+#ifdef HAVE_LIBPCAP
   capture_filter_init();
+#endif
 
   /* the window can be sized only, if it's not already shown, so do it now! */
   main_load_window_geometry(top_level);--- a/gtk/capture_dlg.c
+++ b/gtk/capture_dlg.c
@@ -371,6 +371,7 @@
 
 /** Initialize background capture filter syntax checking
  */
+#ifdef HAVE_LIBPCAP                                                                                                                 
 void capture_filter_init(void) {
   cfc_data.filter_text = NULL;
   cfc_data.filter_te = NULL;
@@ -383,6 +384,7 @@
   g_thread_create(check_capture_filter_syntax, NULL, FALSE, NULL);
 #endif
 }
+#endif
 
 static void
 capture_filter_check_syntax_cb(GtkWidget *w _U_, gpointer user_data _U_)

