--- a/setup.py.orig	2020-09-05 06:36:31.000000000 +0800
+++ b/setup.py	2020-09-14 21:59:49.398923995 +0800
@@ -13,13 +13,13 @@
     vcversioner={},
 
     packages=find_packages(),
-    setup_requires=[
+    tests_require=[
         'pytest-runner',
         'pytest-asyncio; python_version > "3.4"',
         'pytest-trio; python_version > "3.7"',
         'vcversioner'
     ],
-    tests_require=[
+    setup_requires=[
         'twisted',
         'trio; python_version > "3.6"',
         'futures; python_version < "3.0"'
