From 07d88abbfc0056bf02321bdbcd503e59143b56e7 Mon Sep 17 00:00:00 2001
From: Levente Polyak <levente@leventepolyak.net>
Date: Fri, 4 Apr 2025 17:49:22 +0200
Subject: [PATCH] fix(pyproject): declare license file property according to
 PEP

Fixes #72
---
 pyproject.toml | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/pyproject.toml b/pyproject.toml
index 7657122..c7ab099 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -11,11 +11,16 @@ authors = [
 ]
 urls = { "Homepage" = "https://github.com/dirkjanm/ldapdomaindump/" }
 requires-python = ">=3.6"
-license = "MIT"
+license = { file = "LICENSE" }
 dependencies = [
   "dnspython",
   "ldap3>=2.5,!=2.5.2,!=2.5.0,!=2.6"
 ]
+classifiers = [
+    "License :: OSI Approved :: MIT License",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3.12",
+]
 
 [tool.setuptools.packages.find]
 include = ["ldapdomaindump"]
