diff -Naur linux-5.8.7-pentoo/drivers/usb/host/xhci.h linux-5.8.7-pentoo-patched/drivers/usb/host/xhci.h
--- linux-5.8.7-pentoo/drivers/usb/host/xhci.h	2020-08-02 17:21:45.000000000 -0400
+++ linux-5.8.7-pentoo-patched/drivers/usb/host/xhci.h	2020-09-09 13:13:30.873030367 -0400
@@ -1650,15 +1650,15 @@
  * (1K bytes * 8bytes/bit) / (4*32 bits) = 64 segment entries in the table,
  * meaning 64 ring segments.
  * Initial allocated size of the ERST, in number of entries */
-#define	ERST_NUM_SEGS	1
+#define	ERST_NUM_SEGS	3
 /* Initial allocated size of the ERST, in number of entries */
-#define	ERST_SIZE	64
+#define	ERST_SIZE	192
 /* Initial number of event segment rings allocated */
-#define	ERST_ENTRIES	1
+#define	ERST_ENTRIES	3
 /* Poll every 60 seconds */
-#define	POLL_TIMEOUT	60
+#define	POLL_TIMEOUT	120
 /* Stop endpoint command timeout (secs) for URB cancellation watchdog timer */
-#define XHCI_STOP_EP_CMD_TIMEOUT	5
+#define XHCI_STOP_EP_CMD_TIMEOUT	15
 /* XXX: Make these module parameters */
 
 struct s3_save {
