Author: Maxim Levitsky <maximlevitsky@xxxxxxxxx>
Ported: Anton Bolshakov <blshkv@pentoo.xx>

--- a/net/wireless/chan.c
+++ b/net/wireless/chan.c
@@ -492,6 +492,11 @@
 		}
 		return;
 	case NL80211_IFTYPE_MONITOR:
+		if (wdev) {
+			*chan = wdev->channel;
+			*chanmode = CHAN_MODE_SHARED;
+		}
+		return;
 	case NL80211_IFTYPE_AP_VLAN:
 	case NL80211_IFTYPE_WDS:
 		/* these interface types don't really have a channel */
