--- drivers/block/aoe/aoecmd.c.orig	2013-01-02 03:07:24.943072737 +0100
+++ drivers/block/aoe/aoecmd.c	2013-01-02 03:08:23.010068818 +0100
@@ -268,7 +268,7 @@
 }
 
 static void
-fhash(struct frame *f)
+fhash_custom(struct frame *f)
 {
 	struct aoedev *d = f->t->d;
 	u32 n;
@@ -332,7 +332,7 @@
 	skb_put(skb, sizeof *h + sizeof *ah);
 	memset(h, 0, skb->len);
 	f->tag = aoehdr_atainit(d, t, h);
-	fhash(f);
+	fhash_custom(f);
 	t->nout++;
 	f->waited = 0;
 	f->buf = buf;
@@ -449,7 +449,7 @@
 	aoechr_error(buf);
 
 	f->tag = n;
-	fhash(f);
+	fhash_custom(f);
 	h->tag = cpu_to_be32(n);
 	memcpy(h->dst, t->addr, sizeof h->dst);
 	memcpy(h->src, t->ifp->nd->dev_addr, sizeof h->src);
@@ -1224,7 +1224,7 @@
 	skb_put(skb, sizeof *h + sizeof *ah);
 	memset(h, 0, skb->len);
 	f->tag = aoehdr_atainit(d, t, h);
-	fhash(f);
+	fhash_custom(f);
 	t->nout++;
 	f->waited = 0;
 
