diff -Naur rdesktop-1.5.0.orig/xwin.c rdesktop-1.5.0/xwin.c
--- rdesktop-1.5.0.orig/xwin.c	2006-08-07 07:45:44.000000000 -0400
+++ rdesktop-1.5.0/xwin.c	2007-04-16 14:31:44.020671046 -0400
@@ -3219,7 +3219,7 @@
 		return;
 
 	image = XCreateImage(g_display, g_visual, g_depth, ZPixmap, 0,
-			     (char *) data, cx, cy, BitmapPad(g_display), cx * g_bpp / 8);
+			     (char *) data, cx, cy, g_bpp, cx * g_bpp / 8);
 
 	if (g_ownbackstore)
 	{
