--- a/nassl-2.1.0/build_tasks.py.orig	2018-08-27 11:24:25.000000000 +0800
+++ b/nassl-2.1.0/build_tasks.py	2018-08-29 09:56:49.248986644 +0800
@@ -79,16 +79,7 @@
     def fetch_source(self) -> None:
         """Download the tar archive that contains the source code for the library.
         """
-        import requests  # Do not import at the top that this file can be imported by setup.py
-        with TemporaryFile() as temp_file:
-            # Download the source archive
-            request = requests.get(self.src_tar_gz_url)
-            temp_file.write(request.content)
-            # Rewind the file
-            temp_file.seek(0)
-            # Extract the content of the archive
-            tar_file = tarfile.open(fileobj=temp_file)
-            tar_file.extractall(path=_DEPS_PATH)
+        pass
 
     @abstractmethod
     def build(self, ctx: Context) -> None:
