diff -Naur sipvicious-0.2.4.orig/svfphelper.py sipvicious-0.2.4/fphelper.py
--- sipvicious-0.2.4.orig/svfphelper.py	2008-08-10 16:20:18.000000000 +0000
+++ sipvicious-0.2.4/svfphelper.py	2008-12-02 14:58:22.000000000 +0000
@@ -70,7 +70,7 @@
     cPickle.dump(dynamicmatch,output)
     output.close()
 
-def fpdynamicstore(servername,regex,fpfile="totag"):
+def fpdynamicstore(servername,regex,fpfile="/usr/lib/sipvicious/totag"):
     import shelve,os,logging
     log = logging.getLogger('fpdynamicstore')
     if regex is not None:
@@ -82,7 +82,7 @@
         except OSError:
             return
 
-def fpexists(fpname,fpfile="totag"):
+def fpexists(fpname,fpfile="/usr/lib/sipvicious/totag"):
     import shelve,logging
     log = logging.getLogger("fpexists")
     try:
@@ -95,7 +95,7 @@
     dynamicmatch.close()
     return r
 
-def fpdynamic(dyn,fpfile="totag"):
+def fpdynamic(dyn,fpfile="/usr/lib/sipvicious/totag"):
     import shelve
     import logging
     log = logging.getLogger("fpdynamic")
@@ -147,7 +147,7 @@
     orderhash = ''
     return totalhashed,orderhash,hashedheaders
 
-def fpstatic(buffer,fullfn='staticfull',headersfn='staticheaders'):
+def fpstatic(buffer,fullfn='/usr/lib/sipvicious/staticfull',headersfn='/usr/lib/sipvicious/staticheaders'):
     totalhashed,orderhashed,hashedheaders = hashstatic(buffer)
     import shelve
     fulldb = shelve.open(fullfn,writeback=True)
@@ -184,7 +184,7 @@
     return True
 
 
-def fpstore(servername,fullhash,headerhashes,fullfn='staticfull',headersfn='staticheaders'):
+def fpstore(servername,fullhash,headerhashes,fullfn='/usr/lib/sipvicious/staticfull',headersfn='/usr/lib/sipvicious/staticheaders'):
     import shelve,logging
     log = logging.getLogger("fpstore")
     fulldb = shelve.open(fullfn)
@@ -298,7 +298,7 @@
     splitbuff = buff.split(SEP,2)
     return(splitbuff[0])
 
-def groupwherepossible(fpnames,groupdb='groupdb'):
+def groupwherepossible(fpnames,groupdb='/usr/lib/sipvicious/groupdb'):
     import shelve, logging, re
     log = logging.getLogger('grouphwerepossible')
     log.debug("entered")
diff -Naur sipvicious-0.2.4.orig/svhelper.py sipvicious-0.2.4/helper.py
--- sipvicious-0.2.4.orig/svhelper.py	2008-08-10 16:20:18.000000000 +0000
+++ sipvicious-0.2.4/svhelper.py	2008-12-02 14:51:06.000000000 +0000
@@ -818,7 +818,7 @@
         prefix='| ', postfix=' |',wrapfunc=lambda x: wrap_onspace(x,width))
     return o
 
-def outputtoxml(title,labels,db,resdb=None,xsl='sv.xsl'):
+def outputtoxml(title,labels,db,resdb=None,xsl='/usr/lib/sipvicious/sv.xsl'):
     from xml.sax.saxutils import escape
     o  = '<?xml version="1.0" ?>\r\n'
     o += '<?xml-stylesheet type="text/xsl" href="%s"?>\r\n' % escape(xsl)
