diff -Naur scapy-1.1.1-orig/scapy.py scapy-1.1.1/scapy.py
--- scapy-1.1.1-orig/scapy.py	2007-04-09 09:17:35.000000000 -0400
+++ scapy-1.1.1/scapy.py	2009-09-08 17:30:02.000000000 -0400
@@ -3067,9 +3067,9 @@
             if loctrace:
                 trt[trace_id] = loctrace
 
-        tr = map(lambda x: Gnuplot.Data(x,with="lines"), trt.values())
+        tr = map(lambda x: Gnuplot.Data(x,width="lines"), trt.values())
         g = Gnuplot.Gnuplot()
-        world = Gnuplot.File(conf.gnuplot_world,with="lines")
+        world = Gnuplot.File(conf.gnuplot_world,width="lines")
         g.plot(world,*tr)
         return g
 
