diff -urP dsniff-2.4.orig/Makefile.in dsniff-2.4/Makefile.in
--- dsniff-2.4.orig/Makefile.in	2019-01-17 13:41:54.832180094 -0700
+++ dsniff-2.4/Makefile.in	2019-01-17 14:13:30.282057880 -0700
@@ -39,8 +39,8 @@
 X11LIB	= @X_LIBS@ @X_PRE_LIBS@ -lXmu -lX11 @X_EXTRA_LIBS@
 
 INCS	= -I. $(NIDSINC) $(PCAPINC) $(LNETINC) $(DBINC) $(SSLINC) $(X11INC) \
-	  -I$(srcdir)/missing
-LIBS	= @LIBS@ -L$(srcdir) -lmissing
+	  -I$(srcdir)/missing @RPC_CFLAGS@
+LIBS	= @LIBS@ -L$(srcdir) -lmissing @RPC_LIBS@
 
 INSTALL	= @INSTALL@
 INSTALL_PROGRAM = @INSTALL_PROGRAM@
diff -urP dsniff-2.4.orig/configure.in dsniff-2.4/configure.in
--- dsniff-2.4.orig/configure.in	2019-01-17 13:41:54.632180080 -0700
+++ dsniff-2.4/configure.in	2019-01-17 14:15:05.824064731 -0700
@@ -338,4 +338,23 @@
 AC_SUBST(SSLINC)
 AC_SUBST(SSLLIB)
 
+AC_ARG_WITH([libtirpc],
+	   AS_HELP_STRING([--with-libtirpc], [Use libtirpc as RPC implementation (instead of sunrpc)]),
+	   [], [ with_libtirpc=no ])
+
+AS_IF([test "x$with_libtirpc" != xno], 
+      [PKG_CHECK_MODULES([TIRPC], 
+			 [libtirpc], 
+			 [RPC_CFLAGS=$TIRPC_CFLAGS; RPC_LIBS=$TIRPC_LIBS;],
+			[AC_MSG_ERROR([libtirpc requested, but library not found.])]
+		       )], 
+      [AC_CHECK_HEADER(rpc/rpc.h, 
+		      [RPC_CFLAGS=""; RPC_LIBS="";],
+		      [AC_MSG_ERROR([sunrpc requested, but headers are not present.])]
+		     )]
+)
+
+AC_SUBST(RPC_CFLAGS)
+AC_SUBST(RPC_LIBS)
+
 AC_OUTPUT(Makefile)
