--- gencode.c.orig	2013-11-07 16:27:52.026496536 +0800
+++ gencode.c	2013-11-07 16:31:37.379512622 +0800
@@ -7507,6 +7507,18 @@
 			  dir);
 		break;
 
+	case DLT_EN10MB:
+		/* ethernet flags (including direction) are stored
+		* the byte after the 3-byte magic number */
+		if (dir) {
+			/* match outgoing packets */
+			b0 = gen_mcmp(OR_LINK, 3, BPF_B, 1, 0x01);
+		} else {
+			/* incoming packets */
+			b0 = gen_mcmp(OR_LINK, 3, BPF_B, 0, 0x01);
+		}
+		break;
+
 	case DLT_IPNET:
 		if (dir) {
 			/* match outgoing packets */
