diff -urN IntrepidusGroup-mallory-4c3ea86c5679.orig/src/plugin_managers/plugin/image_flip.py IntrepidusGroup-mallory-4c3ea86c5679/src/plugin_managers/plugin/image_flip.py
--- IntrepidusGroup-mallory-4c3ea86c5679.orig/src/plugin_managers/plugin/image_flip.py	2013-01-09 23:48:03.000000000 +0800
+++ IntrepidusGroup-mallory-4c3ea86c5679/src/plugin_managers/plugin/image_flip.py	2014-12-02 17:28:42.923025830 +0800
@@ -1,5 +1,5 @@
 import config
-import Image
+from PIL import Image
 import StringIO
 from base import Base
 
diff -urN IntrepidusGroup-mallory-4c3ea86c5679.orig/src/plugin_managers/plugin/image_invert.py IntrepidusGroup-mallory-4c3ea86c5679/src/plugin_managers/plugin/image_invert.py
--- IntrepidusGroup-mallory-4c3ea86c5679.orig/src/plugin_managers/plugin/image_invert.py	2013-01-09 23:48:03.000000000 +0800
+++ IntrepidusGroup-mallory-4c3ea86c5679/src/plugin_managers/plugin/image_invert.py	2014-12-02 17:29:02.764027204 +0800
@@ -1,6 +1,6 @@
 import config
-import Image
-import ImageChops
+from PIL import Image
+from PIL import ImageChops
 import StringIO
 from base import Base
 
diff -urN IntrepidusGroup-mallory-4c3ea86c5679.orig/src/protocol/http.py IntrepidusGroup-mallory-4c3ea86c5679/src/protocol/http.py
--- IntrepidusGroup-mallory-4c3ea86c5679.orig/src/protocol/http.py	2013-01-09 23:48:03.000000000 +0800
+++ IntrepidusGroup-mallory-4c3ea86c5679/src/protocol/http.py	2014-12-02 17:30:35.984033661 +0800
@@ -10,7 +10,7 @@
 import zlib
 import gzip
 import time
-import Image
+from PIL import Image
 import sys
 import StringIO
 import logging
