diff -Naur hackbar-1.3.2-fx.orig/install.rdf hackbar-1.3.2-fx/install.rdf
--- hackbar-1.3.2-fx.orig/install.rdf	2008-06-18 14:25:02.000000000 +0000
+++ hackbar-1.3.2-fx/install.rdf	2009-01-25 22:17:24.382645243 +0000
@@ -2,10 +2,6 @@
 <RDF:RDF xmlns:em="http://www.mozilla.org/2004/em-rdf#"
          xmlns:NC="http://home.netscape.com/NC-rdf#"
          xmlns:RDF="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
-  <RDF:Description RDF:about="rdf:#$mqLyM3"
-                   em:id="{ec8030f7-c20a-464f-9b0e-13a3a9e97384}"
-                   em:minVersion="1.5"
-                   em:maxVersion="3.0.*" />
   <RDF:Description RDF:about="urn:mozilla:install-manifest"
                    em:id="{F5DDF39C-9293-4d5e-9AA8-E04E6DD5E9B4}"
                    em:name="HackBar"
@@ -14,6 +10,10 @@
                    em:description="A toolbar that helps you find and test SQL injections"
                    em:iconURL="chrome://hackbar/skin/icon.png"
                    em:homepageURL="http://www.izi-services.nl">
+  <RDF:Description RDF:about="rdf:#$mqLyM3"
+                   em:id="{ec8030f7-c20a-464f-9b0e-13a3a9e97384}"
+                   em:minVersion="1.5"
+                   em:maxVersion="3.0.*" />
     <em:targetApplication RDF:resource="rdf:#$mqLyM3"/>
   </RDF:Description>
 </RDF:RDF>
