diff -Nru freeradius-server-2.1.12.orig//src/lib/Makefile freeradius-server-2.1.12/src/lib/Makefile
--- freeradius-server-2.1.12.orig//src/lib/Makefile	2011-09-30 16:12:07.000000000 +0200
+++ freeradius-server-2.1.12/src/lib/Makefile	2011-11-20 12:47:22.000000000 +0100
@@ -54,7 +54,5 @@
 	$(INSTALL) -d -m 755	$(R)$(libdir)
 	$(LIBTOOL) --mode=install $(INSTALL) -c $(TARGET).la \
 		$(R)$(libdir)/$(TARGET).la
-	rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la;
-	ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la
 
 reconfig:
diff -Nru freeradius-server-2.1.12.orig//src/modules/rlm_eap/libeap/Makefile freeradius-server-2.1.12/src/modules/rlm_eap/libeap/Makefile
--- freeradius-server-2.1.12.orig//src/modules/rlm_eap/libeap/Makefile	2011-09-30 16:12:07.000000000 +0200
+++ freeradius-server-2.1.12/src/modules/rlm_eap/libeap/Makefile	2011-11-20 12:47:22.000000000 +0100
@@ -44,5 +44,3 @@
 install: all
 	$(LIBTOOL) --mode=install $(INSTALL) -c $(TARGET).la \
 		$(R)$(libdir)/$(TARGET).la
-	rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la;
-	ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la
diff -Nru freeradius-server-2.1.12.orig//src/modules/rlm_sql/drivers/rules.mak freeradius-server-2.1.12/src/modules/rlm_sql/drivers/rules.mak
--- freeradius-server-2.1.12.orig//src/modules/rlm_sql/drivers/rules.mak	2011-09-30 16:12:07.000000000 +0200
+++ freeradius-server-2.1.12/src/modules/rlm_sql/drivers/rules.mak	2011-11-20 12:47:22.000000000 +0100
@@ -147,6 +147,4 @@
 	if [ "x$(TARGET)" != "x" ]; then \
 	    $(LIBTOOL) --mode=install $(INSTALL) -c \
 		$(TARGET).la $(R)$(libdir)/$(TARGET).la || exit $$?; \
-	    rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la; \
-	    ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la || exit $$?; \
 	fi
diff -Nru freeradius-server-2.1.12.orig//src/modules/rules.mak freeradius-server-2.1.12/src/modules/rules.mak
--- freeradius-server-2.1.12.orig//src/modules/rules.mak	2011-09-30 16:12:07.000000000 +0200
+++ freeradius-server-2.1.12/src/modules/rules.mak	2011-11-20 12:47:22.000000000 +0100
@@ -171,6 +171,4 @@
 	if [ "x$(TARGET)" != "x" ]; then \
 	    $(LIBTOOL) --mode=install $(INSTALL) -c \
 		$(TARGET).la $(R)$(libdir)/$(TARGET).la || exit $$?; \
-	    rm -f $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la; \
-	    ln -s $(TARGET).la $(R)$(libdir)/$(TARGET)-$(RADIUSD_VERSION).la || exit $$?; \
 	fi
