diff -Nru freeradius-server-2.1.6.orig/src/modules/rlm_eap/libeap/Makefile freeradius-server-2.1.6/src/modules/rlm_eap/libeap/Makefile
--- freeradius-server-2.1.6.orig/src/modules/rlm_eap/libeap/Makefile	2009-08-23 10:46:57.000000000 +0200
+++ freeradius-server-2.1.6/src/modules/rlm_eap/libeap/Makefile	2009-08-23 10:47:38.000000000 +0200
@@ -9,6 +9,7 @@
 SRCS		= eapcommon.c eapcrypto.c eapsimlib.c fips186prf.c
 ifneq ($(OPENSSL_LIBS),)
 SRCS		+= cb.c eap_tls.c mppe_keys.c tls.c
+LIBS		+= $(OPENSSL_LIBS)
 endif
 LT_OBJS		= $(SRCS:.c=.lo)
 INCLUDES	= eap_types.h eap_tls.h
