diff -Naur mkxf86config-0.9.10.orig/aticards mkxf86config-0.9.10/aticards
--- mkxf86config-0.9.10.orig/aticards	1970-01-01 01:00:00.000000000 +0100
+++ mkxf86config-0.9.10/aticards	2009-12-03 20:39:33.000000000 +0100
@@ -0,0 +1,31 @@
+Radeon.*HD.*2350
+Radeon.*HD.*2300
+Radeon.*HD.*2400
+Radeon.*HD.*2600
+Radeon.*HD.*2900
+Radeon.*HD.*3430
+Radeon.*HD.*3450
+Radeon.*HD.*3470
+Radeon.*HD.*3690
+Radeon.*HD.*3830
+Radeon.*HD.*3850
+Radeon.*HD.*3870
+Radeon.*HD.*3800
+Radeon.*HD.*4350
+Radeon.*HD.*4300
+Radeon.*HD.*4350
+Radeon.*HD.*4550
+Radeon.*HD.*4500
+Radeon.*HD.*4650
+Radeon.*HD.*4670
+Radeon.*HD.*4600
+Radeon.*HD.*4650
+Radeon.*HD.*4670
+Radeon.*HD.*4830
+Radeon.*HD.*4850
+Radeon.*HD.*4870
+Radeon.*HD.*4890
+Radeon.*HD.*4800
+FireStream
+RV770
+RV670
diff -Naur mkxf86config-0.9.10.orig/mkxf86config.sh mkxf86config-0.9.10/mkxf86config.sh
--- mkxf86config-0.9.10.orig/mkxf86config.sh	2008-04-10 05:27:36.000000000 +0200
+++ mkxf86config-0.9.10/mkxf86config.sh	2009-12-03 22:09:06.000000000 +0100
@@ -53,6 +53,7 @@
 			# We got a resolution on the command line, use it.
 			NEWMODE=$(echo ${x} | cut -d= -f2)
 			RAWMODES="\"${NEWMODE}\""
+		;;
 	esac
 done
 
@@ -80,6 +81,56 @@
 # We create this link since hwsetup cannot properly detect serial mice
 [ -e /dev/mouse ] || ln -sf /dev/ttyS0 /dev/mouse
 
+if [ -e /usr/lib/xorg/modules/drivers/nvidia_drv.so ] \
+	|| [ -e /usr/lib/modules/drivers/nvidia_drv.so ]
+then
+	for x in `cat /etc/X11/nvidiacards`
+	do
+		if [[ ! $(echo ${XDESC} | grep -ie ${x}) == "" ]]
+		then
+			XMODULE="nvidia"
+			break
+		fi
+	done
+fi
+
+if [ -e /usr/lib/xorg/modules/drivers/fglrx_drv.so ] \
+	|| [ -e /usr/lib/modules/drivers/fglrx_drv.so ]
+then
+	for x in `cat /etc/X11/aticards`
+	do
+		if [[ ! $(echo ${XDESC} | grep -ie ${x}) == "" ]]
+		then
+			XMODULE="fglrx"
+			break
+		fi
+	done
+fi
+
+for i in "$@";
+do
+	case "$i" in
+		-nocomposite*)  NOCOMPOSITE="true";;
+		-xvesa*)  XMODULE="vesa";;
+		-xnvidia*)  XMODULE="nvidia";;
+		-xati*)  XMODULE="fglrx";;
+		-xres\=*)
+			# We got a resolution on the command line, use it.
+			NEWMODE=$(echo ${i} | cut -d= -f2)
+			RAWMODES="\"${NEWMODE}\""
+		;;
+		*)
+			echo "Usage   : mkxf86config [options]"
+			echo "Options : -nocomposite disable composite"
+			echo "          -xati        force ati selection"
+			echo "          -xnvidia     force nvidia selection"
+			echo "          -xvesa       force vesa selection"
+			echo "          -xres=x*y    force x resolution"
+			exit 1
+		;;
+	esac
+done
+
 PROTO="${XMOUSETYPE:-Microsoft}"
 NOEMU=""
 [ "${XEMU3}" = "no" ] && NOEMU='s|^.*Emulate3|# No 2 -> 3 Button emulation|g'
@@ -89,11 +140,33 @@
 
 if [ -n "${XMODULE}" ]
 then
-	# Check for Framebuffer X-Modules and initialize framebuffer module
+	# Check for Framebuffer X-Modules and initialize framebuffer module & fix intel
 	case "${XMODULE}" in
 		pvr2fb)
 			modprobe "${XMODULE}" >/dev/null 2>&1
 			XMODULE="fbdev"
+			eselect opengl set xorg-x11
+			;;
+		i810)
+			XMODULE="intel"
+			eselect opengl set xorg-x11
+		;;
+		fglrx)
+			eselect opengl set ati
+		;;
+		nvidia)
+			eselect opengl set nvidia
+		;;
+		vesa)
+			[[ -n $(echo $XDESC | grep -i intel ) ]] && XMODULE="intel"
+			eselect opengl set xorg-x11
+		;;
+		fbdev)
+			[[ -n $(echo $XDESC | grep -i intel ) ]] && XMODULE="intel"
+			eselect opengl set xorg-x11
+		;;
+		*)
+			eselect opengl set xorg-x11
 		;;
 	esac
 fi
@@ -142,8 +215,8 @@
 # working support for alpha/amd64/ppc/x86 for the 2007.0 Gentoo release.  If
 # anyone has more reliable, cross-platform methods, I'm all ears.
 #RAWMODES=$(ddcxinfo-knoppix -monitor | grep ModeLine | sed -r "s/.*\"([0-9]+x[0-9]+)\".*/\1/g"| sort -rg | uniq | xargs echo | sed -r "s/([0-9]+x[0-9]+)/\"\1\"/g")
-[ -z "${RAWMODES}" ] && RAWMODES="\"1024x768\" \"800x600\" \"640x480\""
-MODES="Modes ${RAWMODES}"
+MODES="#We let X11 auto-detect"
+[ -n "${RAWMODES}" ] && MODES="Modes ${RAWMODES}"
 
 # We need to check this because serial mice have long timeouts
 SERIALMOUSE="$(ls -l1 /dev/mouse* 2>/dev/null | awk '/ttyS/{print $NF ; exit 0}')"
@@ -194,6 +267,15 @@
 	else
 		SYNMOUSE='s|^.*InputDevice.*"Synaptics".*$|#No Synaptics touchpad found|g;'
 	fi
+	CHECK=$(cat /proc/bus/input/devices | grep -i alps | wc -l)
+	if [ ${CHECK} -gt 0 ]
+	then
+		modprobe -q evdev
+		ALPSDEV=/dev/input/$(cat /proc/bus/input/devices | egrep -i -A 5 "^N: .*alpsps/2.*" | grep Handlers | sed -r "s/.*(event[0-9]+).*/\1/g")
+		ALPSMOUSE=""
+	else
+		ALPSMOUSE='s|^.*InputDevice.*"AlpsPad".*$|#No Alps touchpad found|g;'
+	fi
 fi
 
 # Write Monitor data now
@@ -210,8 +292,12 @@
 	if [ -e /usr/lib/xorg/modules/input/vmmouse_drv.so ] || \
 	[ -e /usr/lib/modules/input/vmmouse_drv.so ]
 	then
-		MOUSEDRIVER='s|^.*Driver.*"mouse".*$|\tDriver\t"vmmouse"|g;'
+		# MOUSEDRIVER='s|^.*Driver.*"mouse".*$|\tDriver\t"vmmouse"|g;'
+		ewarn "Not using the vmmouse driver as there are some issues"
+		ewarn "See http://communities.vmware.com/message/716756"
 	fi
+	PSMOUSE='s|^.*InputDevice.*"PS/2 Mouse".*$|# PS/2 Mouse using /dev/input/mice in Kernel 2.6|g;'
+	MODES="Modes \"1024x768\" \"800x600\" \"640x480\""
 fi
 
 #VirtualPC special handline
@@ -220,6 +306,17 @@
 then
 	VPC='s|^.*BusID.*PCI.*$|BusID "PCI:0:8:0"|g;'
 	DEPTH='s|DefaultColorDepth 24|DefaultColorDepth 16|g;'
+	MODES="Modes \"1024x768\" \"800x600\" \"640x480\""
+fi
+
+#VirtualPC special handline
+VBOX="$(echo ${XDESC} | grep -oi virtualbox)"
+if [ -n "${VBOX}" ]
+then
+	XMODULE="vboxvideo"
+	DEPTH='s|DefaultColorDepth 24|DefaultColorDepth 16|g;'
+	PSMOUSE='s|^.*InputDevice.*"PS/2 Mouse".*$|# PS/2 Mouse using /dev/input/mice in Kernel 2.6|g;'
+	MODES="Modes \"1024x768\" \"800x600\" \"640x480\""
 fi
 
 # If we don't have a XMODULE set, use fbdev as fall-back
@@ -235,10 +332,10 @@
 SWCURSOR=""
 MONITORLAYOUT=""
 case "${XMODULE}" in
-	ati|nv|trident)
+	ati|nv|trident|nvidia)
 		SWCURSOR='s|^.*#Option.*"sw_cursor".*$|Option "sw_cursor"|g;'
 	;;
-	radeon)
+	radeon|fglrx)
 		SWCURSOR='s|^.*#Option.*"sw_cursor".*$|Option "sw_cursor"|g;'
 		MONITORLAYOUT='s|^.*#Option.*"MonitorLayout".*$|Option "MonitorLayout"|g;'
 	;;
@@ -249,14 +346,24 @@
 NOPM=""
 DPMS=""
 
+if [ -n "$NOCOMPOSITE" ]; then
+	NOCOMPOSITE='/Option.*"Composite"/ s|Enable|Disable|;'
+else
+	NOCOMPOSITE=''
+fi
+
 #checkbootparam noapm && NOPM='Option	"NoPM"	"true"' || DPMS='Option	"DPMS"	"true"'
 
+# We are oin 2.6 let's remove PSMOUSE as it messes the xorg input
+PSMOUSE='s|^.*InputDevice.*"PS/2 Mouse".*$|# PS/2 Mouse using /dev/input/mice in Kernel 2.6|g;'
+
 sed -e 's|@@PROTOCOL@@|'"${PROTO}"'|g;'"${NOEMU}" \
     -e '/@@MONITOR@@/r '"${MONITORTMP}" \
     -e 's|@@MONITOR@@||g' \
     -e 's|@@NOPM@@|'"${NOPM}"'|g' \
-    -e 's|@@XMODULE@@|'"${XMODULE}"'|g;'"${VMWARE}""${VPC}""${SERIALMOUSE}""${USBMOUSE}""${PSMOUSE}""${SWCURSOR}""${MONITORLAYOUT}""${WHEEL}""${SYNMOUSE}""${MOUSEDRIVER}" \
+    -e 's|@@XMODULE@@|'"${XMODULE}"'|g;'"${NOCOMPOSITE}""${VMWARE}""${VPC}""${SERIALMOUSE}""${USBMOUSE}""${PSMOUSE}""${SWCURSOR}""${MONITORLAYOUT}""${WHEEL}""${SYNMOUSE}""${ALPSMOUSE}""${MOUSEDRIVER}" \
     -e 's|@@SYNDEV@@|'"${SYNDEV}"'|g' \
+    -e 's|@@ALPSDEV@@|'"${ALPSDEV}"'|g' \
     -e 's|@@MODES@@|'"${MODES}"'|g;'"${DEPTH}" \
     -e 's|"XkbLayout" *"[^"]*"|"XkbLayout" "'"${XKEYBOARD}"'"|g;'"${DEADKEYS}" \
     /etc/X11/xorg.conf.in >/etc/X11/xorg.conf
diff -Naur mkxf86config-0.9.10.orig/nvidiacards mkxf86config-0.9.10/nvidiacards
--- mkxf86config-0.9.10.orig/nvidiacards	1970-01-01 01:00:00.000000000 +0100
+++ mkxf86config-0.9.10/nvidiacards	2009-12-03 20:39:33.000000000 +0100
@@ -0,0 +1,89 @@
+GeForce.*GTX.*295
+GeForce.*GTX.*285
+GeForce.*GTX.*280
+GeForce.*GTX.*275
+GeForce.*GTX.*260
+GeForce.*GTS.*250
+GeForce.*GTS.*150
+GeForce.*GT.*130
+GeForce.*GT.*120
+GeForce.*G100
+GeForce.*9800.*GX2
+GeForce.*9800.*GTX+
+GeForce.*9800.*GTX
+GeForce.*9800.*GT
+GeForce.*9600.*GSO
+GeForce.*9600.*GT
+GeForce.*9500.*GT
+GeForce.*9400.*GT
+GeForce.*8800.*Ultra
+GeForce.*8800.*GTX
+GeForce.*8800.*GTS
+GeForce.*8800.*GT
+GeForce.*8800.*GS
+GeForce.*8600.*GTS
+GeForce.*8600.*GT
+GeForce.*8500.*GT
+GeForce.*8400.*GS
+GeForce.*9400.*mGPU
+GeForce.*9300.*mGPU
+GeForce.*8300.*mGPU
+GeForce.*8200.*mGPU
+GeForce.*8100.*mGPU
+Tesla.*S1070
+Tesla.*C1060
+Tesla.*C870
+Tesla.*D870
+Tesla.*S870
+Quadro.*FX.*5800
+Quadro.*FX.*5600
+Quadro.*FX.*4800
+Quadro.*FX.*4700.*X2
+Quadro.*FX.*4600
+Quadro.*FX.*3700
+Quadro.*FX.*1700
+Quadro.*FX.*570
+Quadro.*FX.*470
+Quadro.*FX.*370
+Quadro.*NVS.*290
+Quadro.*Plex.*2100.*D4
+Quadro.*Plex.*2200.*D2
+Quadro.*Plex.*2100.*S4
+Quadro.*Plex.*1000.*Model.*IV
+GeForce.*GT.*130M
+GeForce.*G110M
+GeForce.*G105M
+GeForce.*9800M.*GTX
+GeForce.*9800M.*GTS
+GeForce.*9800M.*GT
+GeForce.*9700M.*GTS
+GeForce.*9700M.*GT
+GeForce.*9650M.*GS
+GeForce.*9600M.*GT
+GeForce.*9600M.*GS
+GeForce.*9500M.*GS
+GeForce.*9500M.*G
+GeForce.*9300M.*GS
+GeForce.*9300M.*G
+GeForce.*9200M.*GS
+GeForce.*9100M.*G
+GeForce.*8800M.*GTS
+GeForce.*8700M.*GT
+GeForce.*8600M.*G
+GeForce.*8400M.*G
+Quadro.*FX.*3700M
+Quadro.*FX.*3600M
+Quadro.*FX.*2700M
+Quadro.*FX.*1700M
+Quadro.*FX.*1600M
+Quadro.*FX.*770M
+Quadro.*FX.*570M
+Quadro.*FX.*370M
+Quadro.*FX.*360M
+Quadro.*NVS.*320M
+Quadro.*NVS.*160M
+Quadro.*NVS.*150M
+Quadro.*NVS.*140M
+Quadro.*NVS.*135M
+Quadro.*NVS.*130M
+
diff -Naur mkxf86config-0.9.10.orig/xorg.conf.in mkxf86config-0.9.10/xorg.conf.in
--- mkxf86config-0.9.10.orig/xorg.conf.in	2008-04-09 02:28:56.000000000 +0200
+++ mkxf86config-0.9.10/xorg.conf.in	2009-12-03 20:39:33.000000000 +0100
@@ -6,6 +6,7 @@
 #	InputDevice	"Serial Mouse" "AlwaysCore"
 	InputDevice	"USB Mouse" "AlwaysCore"
 	InputDevice	"Synaptics" "AlwaysCore"
+	InputDevice	"AlpsPad" "AlwaysCore"
 EndSection
 
 Section "ServerFlags"
@@ -13,6 +14,10 @@
 	@@NOPM@@
 EndSection
 
+Section "Extensions"
+	Option	"Composite" "Enable"
+EndSection
+
 Section "Files"
 	FontPath	"/usr/share/fonts/util"
 	FontPath	"/usr/share/fonts/encodings"
@@ -23,9 +28,10 @@
 	FontPath	"/usr/local/share/fonts"
 	FontPath	"/usr/share/fonts/default"
 	FontPath	"/usr/share/fonts/TTF"
-	FontPath	"/usr/share/fonts/type1"
+	FontPath	"/usr/share/fonts/Type1"
 	FontPath	"/usr/share/fonts/100dpi"
 	FontPath	"/usr/share/fonts/75dpi"
+	FontPath	"/usr/share/fonts/ttf-bitsream-vera"
 	FontPath	"/usr/share/fonts/arphicfonts"
 	FontPath	"/usr/share/fonts/jisx0213"
 	FontPath	"/usr/share/fonts/shinonome"
@@ -105,7 +111,7 @@
 	Identifier	"PS/2 Mouse"
 	Driver	"mouse"
 	Option	"Protocol" "IMPS/2"
-	Option	"Device" "/dev/misc/psaux"
+	Option	"Device" "/dev/psaux"
 	Option	"Emulate3Buttons" "true"
 	Option	"Emulate3Timeout" "70"
 	Option	"SendCoreEvents"  "true"
@@ -136,10 +142,46 @@
 	Option	"MaxTapTime" "180"
 	Option	"MaxTapMove" "220"
 	Option	"VertScrollDelta" "100"
-	Option	"MinSpeed" "0.02"
-	Option	"MaxSpeed" "0.10"
-	Option	"AccelFactor" "0.0010"
+	Option	"MinSpeed" "0.3"
+	Option	"MaxSpeed" "0.50"
+	Option	"AccelFactor" "0.080"
+	Option	"SHMConfig" "on"
+	Option	"TapButton1" "1"
+	Option	"TapButton2" "2"
+	Option	"TapButton3" "3"
+EndSection
+
+Section "InputDevice"
+	Identifier	"AlpsPad"
+	Driver	"synaptics"
+	Option	"Device" "@@ALPSDEV@@"
+	Option	"Protocol" "event"
+	Option	"LeftEdge" "130"
+	Option	"RightEdge" "840"
+	Option	"TopEdge" "130"
+	Option	"BottomEdge" "640"
+	Option	"FingerLow" "7"
+	Option	"FingerHigh" "8"
+	Option	"MaxTapTime" "180"
+	Option	"MaxTapMove" "110"
+	Option	"EmulateMidButtonTime" "75"
+	Option	"VertScrollDelta"	"20"
+	Option	"HorizScrollDelta" "20"
+	Option	"MinSpeed" "0.60"
+	Option	"MaxSpeed" "1.10"
+	Option	"AccelFactor" "0.030"
+	Option	"EdgeMotionMinSpeed" "200"
+	Option	"EdgeMotionMaxSpeed" "200"
+	Option	"UpDownScrolling" "1"
+	Option	"CircularScrolling" "1"
+	Option	"CircScrollDelta" "0.1"
+	Option	"CircScrollTrigger" "2"
 	Option	"SHMConfig" "on"
+	Option	"Emulate3Buttons" "on"
+        Option	"SendCoreEvents" "true"
+	Option	"TapButton1" "1"
+	Option	"TapButton2" "2"
+	Option	"TapButton3" "3"
 EndSection
 
 # Auto-generated by mkxf86config
@@ -162,6 +204,11 @@
 	VendorName  "All"
 	BoardName   "All"
 #	BusID       "PCI:1:0:0"
+
+	Option "XAANoOffscreenPixmaps"
+	Option "AllowGLXWithComposite" "true"
+	Option "EnablePageFlip" "true"
+	Option "TripleBuffer" "true"
 EndSection
 
 Section "Screen"
@@ -169,6 +216,8 @@
 	Device	"Card0"
 	Monitor	"Monitor0"
 	DefaultColorDepth 24
+	Option "AddARGBGLXVisuals" "true"
+	Option "DisableGLXRootClipping" "true"
 	SubSection "Display"
 		Depth	1
 		@@MODES@@
