diff -Naur mkxf86config-0.9.11.orig/mkxf86config.sh mkxf86config-0.9.11/mkxf86config.sh
--- mkxf86config-0.9.11.orig/mkxf86config.sh	2010-11-24 11:36:29.000000000 +0100
+++ mkxf86config-0.9.11/mkxf86config.sh	2011-01-27 22:36:46.000000000 +0100
@@ -1,7 +1,10 @@
 #!/bin/bash
+# Some stuff got taken from mkxorgconfig made by
+# (C) Klaus Knopper 2006
+# License: GPL V2
 
 source /etc/profile
-source /sbin/functions.sh
+source /lib/rc/sh/functions.sh
 
 # First, get the command line
 CMDLINE="$(</proc/cmdline)"
@@ -57,6 +60,12 @@
 	esac
 done
 
+#Ripped from knoppix
+checkbootparam(){
+	case "$CMDLINE" in *\ $1*) return 0;; esac
+	return 1
+}
+
 TMPFILE="/tmp/mkxf86config-$$"
 MONITORTMP="${TMPFILE}-monitor"
 
@@ -165,6 +174,8 @@
 			eselect opengl set xorg-x11
 		;;
 	esac
+else
+	eselect opengl set xorg-x11
 fi
 
 # We used to use ddcxinfo-knoppix for monitor information, now we will just let
@@ -305,7 +316,7 @@
 	MODES="Modes \"1024x768\" \"800x600\" \"640x480\""
 fi
 
-#VirtualPC special handline
+#Vbox special handline
 VBOX="$(echo ${XDESC} | grep -oi virtualbox)"
 if [ -n "${VBOX}" ]
 then
@@ -328,16 +339,12 @@
 SWCURSOR=""
 MONITORLAYOUT=""
 case "${XMODULE}" in
-	ati|nv|trident|nvidia)
-		SWCURSOR='s|^.*#Option.*"sw_cursor".*$|Option "sw_cursor"|g;'
-	;;
-	radeon|fglrx)
+	trident)
 		SWCURSOR='s|^.*#Option.*"sw_cursor".*$|Option "sw_cursor"|g;'
-		MONITORLAYOUT='s|^.*#Option.*"MonitorLayout".*$|Option "MonitorLayout"|g;'
 	;;
 esac
 
-# We must use NoPM, because some machines freeze if Power management is beingi
+# We must use NoPM, because some machines freeze if Power management is being
 # activated.
 NOPM=""
 DPMS=""
@@ -348,15 +355,16 @@
 	NOCOMPOSITE=''
 fi
 
-#checkbootparam noapm && NOPM='Option	"NoPM"	"true"' || DPMS='Option	"DPMS"	"true"'
+checkbootparam noapm && NOPM='Option	"NoPM"	"true"' || DPMS='Option	"DPMS"	"true"'
 
-# We are oin 2.6 let's remove PSMOUSE as it messes the xorg input
+# We are in 2.6 let's remove PSMOUSE as it messes the xorg input
 PSMOUSE='s|^.*InputDevice.*"PS/2 Mouse".*$|# PS/2 Mouse using /dev/input/mice in Kernel 2.6|g;'
 
 sed -e 's|@@PROTOCOL@@|'"${PROTO}"'|g;'"${NOEMU}" \
     -e '/@@MONITOR@@/r '"${MONITORTMP}" \
     -e 's|@@MONITOR@@||g' \
     -e 's|@@NOPM@@|'"${NOPM}"'|g' \
+    -e 's|@@DPMS@@|'"${DPMS}"'|g' \
     -e 's|@@XMODULE@@|'"${XMODULE}"'|g;'"${NOCOMPOSITE}""${VMWARE}""${VPC}""${SERIALMOUSE}""${USBMOUSE}""${PSMOUSE}""${SWCURSOR}""${MONITORLAYOUT}""${WHEEL}""${SYNMOUSE}""${ALPSMOUSE}""${MOUSEDRIVER}" \
     -e 's|@@SYNDEV@@|'"${SYNDEV}"'|g' \
     -e 's|@@ALPSDEV@@|'"${ALPSDEV}"'|g' \
diff -Naur mkxf86config-0.9.11.orig/xorg.conf.in mkxf86config-0.9.11/xorg.conf.in
--- mkxf86config-0.9.11.orig/xorg.conf.in	2010-11-24 11:14:51.000000000 +0100
+++ mkxf86config-0.9.11/xorg.conf.in	2011-01-27 21:53:58.000000000 +0100
@@ -11,8 +11,10 @@
 
 Section "ServerFlags"
 	Option	"AllowMouseOpenFail" "true"
+	# Prevent HAL from breaking this config
 	Option	"AutoAddDevices" "false"
 	@@NOPM@@
+	@@DPMS@@
 EndSection
 
 Section "Extensions"
@@ -207,10 +209,15 @@
 	BoardName   "All"
 #	BusID       "PCI:1:0:0"
 
+# compiz, beryl 3D-Support with DRI & Composite
 	Option "XAANoOffscreenPixmaps"
 	Option "AllowGLXWithComposite" "true"
 	Option "EnablePageFlip" "true"
 	Option "TripleBuffer" "true"
+
+# Tweaks for the xorg 7.4 (otherwise broken) "intel" driver
+	Option "Tiling" "no"
+	Option "Legacy3D" "false"
 EndSection
 
 Section "Screen"
@@ -253,3 +260,10 @@
 Section "DRI"
 	Mode 0666
 EndSection
+
+Section "InputClass"
+	Identifier  "Keyboard Defaults"
+	MatchIsKeyboard "yes"
+	Option      "XkbLayout" ""
+EndSection
+
