Description: Fix typo in config.h
Author: Uwe Hermann <uwe@debian.org>
Last-Update: 2018-02-20

--- btscanner-2.1.orig/config.h.in
+++ btscanner-2.1/config.h.in
@@ -118,5 +118,5 @@
 /* Define to rpl_malloc if the replacement function should be used. */
 #undef malloc
 
-/* Define to `unsigned' if <sys/types.h> does not define. */
+/* Define to `unsigned int' if <sys/types.h> does not define. */
 #undef size_t
