From 3554dfd23c0c1080a0a3152265802410db6dd24e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Diego=20Elio=20Petten=C3=B2?= <flameeyes@flameeyes.eu>
Date: Sun, 19 Aug 2012 19:21:12 -0700
Subject: [PATCH 2/7] Respect DESTDIR for pythoninstall.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit


Signed-off-by: Diego Elio Pettenò <flameeyes@flameeyes.eu>
---
 Makefile.in | 2 +-
 1 file modificato, 1 inserzione(+). 1 rimozione(-)

diff --git a/Makefile.in b/Makefile.in
index 4d00168..cb37410 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -229,7 +229,7 @@ pythonmodules: subdirs
         fi
 
 pythoninstall:
-	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir) ; \
+	@(dir=`pwd`; cd python; $(PYMAKE) install --basedir=$$dir --root=$(DESTDIR)) ; \
         if test $$? != 0 ; then \
            exit 1 ; \
         fi
-- 
1.7.12

