--- include/ec_file.h	2010-04-16 00:35:14.093755063 +0200
+++ include/ec_file.h	2010-04-16 00:34:25.133751660 +0200
@@ -11,7 +11,7 @@
 #define MAC_FINGERPRINTS   "etter.finger.mac"
 #define TCP_FINGERPRINTS   "etter.finger.os"
 #define SERVICES_NAMES     "etter.services"
-#define ETTER_CONF         "etter.conf"
+#define ETTER_CONF         "ucsniff.conf"
 #define ETTER_DNS          "etter.dns"
 #define ETTER_FIELDS       "etter.fields"
 #define CERT_FILE          "etter.ssl.crt"
--- share/Makefile.am.orig	2011-06-03 07:08:54.000000000 +0800
+++ share/Makefile.am	2012-03-28 13:52:40.853712634 +0800
@@ -17,8 +17,8 @@
                etterfilter.cnt \
                etterfilter.tbl \
                etterlog.dtd \
-	            ettercap.png
+               ettercap.png
 
-sysconf_DATA = etter.conf
+sysconf_DATA = ucsniff.conf
 
 EXTRA_DIST = etter*
