--- a/setup.py	2018-08-23 07:36:33.812024967 -0500
+++ b/setup.py	2018-08-23 07:36:56.878690371 -0500
@@ -5,25 +5,25 @@
 except ImportError:
     from distutils.core import setup
 
-try:
-    from pip._internal.req import parse_requirements
-except ImportError:
-    from pip.req import parse_requirements
+#try:
+#    from pip._internal.req import parse_requirements
+#except ImportError:
+#    from pip.req import parse_requirements
 
 from distutils.core import Extension
 import plasma
 
-requirements = parse_requirements('requirements.txt', session=False)
+#requirements = parse_requirements('requirements.txt', session=False)
 
-requires = []
-for item in requirements:
+#requires = []
+#for item in requirements:
     # we want to handle package names and also repo urls
-    if getattr(item, 'url', None):  # older pip has url
-        links.append(str(item.url))
-    if getattr(item, 'link', None): # newer pip has link
-        links.append(str(item.link))
-    if item.req:
-        requires.append(str(item.req))
+#    if getattr(item, 'url', None):  # older pip has url
+#        links.append(str(item.url))
+#    if getattr(item, 'link', None): # newer pip has link
+#        links.append(str(item.link))
+#    if item.req:
+#        requires.append(str(item.req))
 
 
 x86_analyzer = Extension('plasma.lib.arch.x86.analyzer',
@@ -59,7 +59,7 @@
               'plasma.scripts',
     ],
     package_dir={'plasma':'plasma'},
-    install_requires=requires,
+#    install_requires=requires,
     entry_points = {
         "console_scripts": [
             "plasma = plasma.main:console_entry",
