diff -urN aiocmd-0.1.2.orig/aiocmd/aiocmd.py aiocmd-0.1.2/aiocmd/aiocmd.py
--- aiocmd-0.1.2.orig/aiocmd/aiocmd.py	2019-09-12 20:35:08.000000000 +0800
+++ aiocmd-0.1.2/aiocmd/aiocmd.py	2020-04-21 11:06:58.788721853 +0800
@@ -11,7 +11,6 @@
     from prompt_toolkit.completion.nested import NestedCompleter
 except ImportError:
     from aiocmd.nested_completer import NestedCompleter
-from prompt_toolkit.eventloop.defaults import use_asyncio_event_loop
 
 
 class ExitPromptException(Exception):
@@ -34,7 +33,6 @@
     aliases = {"?": "help", "exit": "quit"}
 
     def __init__(self, ignore_sigint=True):
-        use_asyncio_event_loop()
         self.completer = self._make_completer()
         self.session = None
         self._ignore_sigint = ignore_sigint
@@ -55,7 +53,7 @@
     async def _run_prompt_forever(self):
         while True:
             try:
-                result = await self.session.prompt(self.prompt, async_=True, completer=self.completer)
+                result = await self.session.prompt_async(self.prompt, completer=self.completer)
             except EOFError:
                 return
 
