diff -Naur CUDA-Multiforcer-Release.orig/Makefile CUDA-Multiforcer-Release/Makefile
--- CUDA-Multiforcer-Release.orig/Makefile	2009-05-13 17:13:04.000000000 +0200
+++ CUDA-Multiforcer-Release/Makefile	2009-05-13 20:44:40.000000000 +0200
@@ -33,7 +33,9 @@
 #
 ################################################################################
 # Add source files here
-EXECUTABLE	:= CUDA-Multiforcer
+ROOTDIR		:= /opt/cuda/sdk/
+ROOTBINDIR	:= ./bin
+EXECUTABLE	:= cuda-multiforcer
 # Cuda source files (compiled with cudacc)
 CUFILES		:= Multiforcer.cu
 # C/C++ source files (compiled with gcc / c++)
@@ -44,4 +46,4 @@
 # Rules and targets
 
 
-include ../../common/common.mk
+include /opt/cuda/sdk/common/common.mk
diff -Naur CUDA-Multiforcer-Release.orig/Multiforcer.cu CUDA-Multiforcer-Release/Multiforcer.cu
--- CUDA-Multiforcer-Release.orig/Multiforcer.cu	2009-05-13 17:13:10.000000000 +0200
+++ CUDA-Multiforcer-Release/Multiforcer.cu	2009-05-13 20:44:01.000000000 +0200
@@ -1,4 +1,4 @@
-#include "./argtable2-9/src/argtable2.h"
+#include <argtable2.h>
 #include <cuda.h>
 #include <cutil.h>
 #include "Multiforcer.h"
