From 5783aaed9e318853f98857633f647c96d5142e42 Mon Sep 17 00:00:00 2001
From: William Hubbs <w.d.hubbs@gmail.com>
Date: Sun, 18 Oct 2009 17:29:58 -0500
Subject: [PATCH] fix typo

---
 init.d/network.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/init.d/network.in b/init.d/network.in
index 5e60963..7f64b17 100644
--- a/init.d/network.in
+++ b/init.d/network.in
@@ -174,7 +174,7 @@ start()
 {
 	local cr=0 r= int= intv= cmd= args= upcmd=
 
-	if [ -z "$domainname" && -s /etc/defaultdomain ]; then
+	if [ -z "$domainname" -a -s /etc/defaultdomain ]; then
 		domainname=$(cat /etc/defaultdomain)
 	fi
 	if [ -n "$domainname" ]; then
-- 
1.6.4.4

