diff -Naur proxmark3-3.0.1/client/Makefile proxmark3-3.0.1-fixed/client/Makefile
--- proxmark3-3.0.1/client/Makefile	2017-06-09 13:50:55.000000000 +0000
+++ proxmark3-3.0.1-fixed/client/Makefile	2018-08-11 23:53:24.594635301 +0000
@@ -4,9 +4,9 @@
 # the license.
 #-----------------------------------------------------------------------------
 
-CC = gcc
-CXX = g++
-LD = g++
+CC ?= gcc
+CXX ?= g++
+LD = g++
 TAR = tar
 TARFLAGS = -C .. --ignore-failed-read -rvf
 RM = rm -f
@@ -18,9 +18,9 @@
 
 LDLIBS = -L/opt/local/lib -L/usr/local/lib -lreadline -lpthread -lm
 LUALIB = ../liblua/liblua.a
-LDFLAGS = $(COMMON_FLAGS)
-CFLAGS = -std=c99 -D_ISOC99_SOURCE -I. -I../include -I../common -I../zlib -I/opt/local/include -I../liblua -Wall $(COMMON_FLAGS) -g -O3
-CXXFLAGS = -I../include -Wall -O3
+#LDFLAGS := $(LDFLAGS)
+CFLAGS := -std=c99 -D_ISOC99_SOURCE -I. -I../include -I../common -I../zlib -I/opt/local/include -I../liblua $(CFLAGS)
+CXXFLAGS := -I../include $(CXXFLAGS)
 
 LUAPLATFORM = generic
 platform = $(shell uname)
