diff -ur a/bob_client/format_mysql/format_mysql.c b/bob_client/format_mysql/format_mysql.c
--- a/bob_client/format_mysql/format_mysql.c	2005-10-28 17:23:07.000000000 +0400
+++ b/bob_client/format_mysql/format_mysql.c	2019-11-10 17:28:21.471512052 +0300
@@ -27,6 +27,7 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <math.h>
 #include "params.h"
 #ifdef HAVE_MATH_H
 #include <math.h>
diff -ur a/bob_client/timer.c b/bob_client/timer.c
--- a/bob_client/timer.c	2006-06-28 16:16:57.000000000 +0400
+++ b/bob_client/timer.c	2019-11-10 17:26:09.991841328 +0300
@@ -4,6 +4,7 @@
 #include <sys/time.h>
 #include <signal.h>
 #include <stdio.h>
+#include <stdlib.h>
 
 #include <time.h>
 #ifdef HAVE_LIBDMALLOC
diff -ur a/compat/get_full_login.c b/compat/get_full_login.c
--- a/compat/get_full_login.c	2005-11-30 16:58:32.000000000 +0300
+++ b/compat/get_full_login.c	2019-11-10 17:29:45.199569756 +0300
@@ -1,5 +1,6 @@
 #include <unistd.h>
 #include <string.h>
+#include <stdio.h>
 
 #define USER_HOST_SEP	'@'
 #define BUFSIZE		1024
diff -ur a/librw/str_funcs.c b/librw/str_funcs.c
--- a/librw/str_funcs.c	2005-12-06 16:26:51.000000000 +0300
+++ b/librw/str_funcs.c	2019-11-10 17:34:57.303411231 +0300
@@ -2,12 +2,7 @@
 #include "config_types.h"
 
 #include <stdio.h>
-#ifdef HAVE_LIBDMALLOC
-# include <dmalloc.h>
-#else
-# include <stdlib.h>
-#endif
-
+#include <stdlib.h>
 #include <string.h>
 
 
