diff -ur a/theHarvester/__main__.py b/theHarvester/__main__.py
--- a/theHarvester/__main__.py	2019-10-14 23:48:56.000000000 +0300
+++ b/theHarvester/__main__.py	2020-03-02 12:46:37.460953085 +0300
@@ -489,7 +489,11 @@
         print('---------------------')
         all_hosts = sorted(list(set(all_hosts)))
         full_host = hostchecker.Checker(all_hosts)
-        full, ips = asyncio.run(full_host.check())
+        if sys.version_info[1] <= 6:
+            loop = asyncio.get_event_loop()
+            full, ips = loop.run_until_complete(full_host.check())
+        else:
+            full, ips = asyncio.run(full_host.check())
         db = stash.stash_manager()
         for host in full:
             host = str(host)
