--- a/spike.c	2020-02-29 18:27:41.677896000 +0100
+++ b/spike.c	2020-02-29 18:28:07.356896000 +0100
@@ -1082,7 +1082,7 @@
       if (current_spike->fd==-1)
 	{
 	  printf("tried to send to a closed socket!\n");
-	  return 0;
+	  return 1;
 	}
       c=s_get_databuf();
       totalsize=s_get_size();
@@ -1160,7 +1160,7 @@
       if (current_spike->fd==-1)
 	{
 	  printf("tried to send to a closed socket!\n");
-	  return 0;
+	  return 1;
 	}
       alarm(5);
       signal(SIGALRM,(void*)sighandler);
