diff -ur a/Makefile b/Makefile
--- a/Makefile	2014-04-16 06:49:14.000000000 +0400
+++ b/Makefile	2019-10-27 00:15:56.765813375 +0300
@@ -37,19 +37,13 @@
 	ldid -S dnstest
 
 install: all
-	mkdir -p /usr/local/bin
-	cp nbquery    /usr/local/bin/nbquery
-	cp nbsniff    /usr/local/bin/nbsniff
-	cp dnsxss     /usr/local/bin/dnsxss
-	cp dnslogger  /usr/local/bin/dnslogger
-	cp dnscat     /usr/local/bin/dnscat 
-	cp dnstest    /usr/local/bin/dnstest
-	chown root.root /usr/local/bin/nbquery
-	chown root.root /usr/local/bin/nbsniff
-	chown root.root /usr/local/bin/dnsxss
-	chown root.root /usr/local/bin/dnslogger
-	chown root.root /usr/local/bin/dnscat
-	chown root.root /usr/local/bin/dnstest
+	mkdir -p $(DESTDIR)/usr/bin
+	cp nbquery    $(DESTDIR)/usr/bin/nbquery
+	cp nbsniff    $(DESTDIR)/usr/bin/nbsniff
+	cp dnsxss     $(DESTDIR)/usr/bin/dnsxss
+	cp dnslogger  $(DESTDIR)/usr/bin/dnslogger
+	cp dnscat     $(DESTDIR)/usr/bin/dnscat 
+	cp dnstest    $(DESTDIR)/usr/bin/dnstest
 
 remove:
 	rm -f /usr/local/bin/nbquery
