diff --git a/vmware-player-extras.py b/vmware-player-extras.py
index 21595b1..e8adf08 100644
--- a/vmware-player-extras.py
+++ b/vmware-player-extras.py
@@ -6,7 +6,8 @@ VMware Player Extras component installer.
 
 DEST = LIBDIR/'vmware'
 SETTINGS = { 'vmware.fullpath': BINDIR/'vmware', }
-CONF = DEST/'setup/vmware-config'
+import os
+CONF = path(os.environ['WORKDIR'])/'vmware-config.sh'
 
 
 class PlayerExtras(Installer):
