diff --exclude-from=/home/dang/.diffrc -u -ruN gdm-2.13.0.1.orig/configure gdm-2.13.0.1/configure
--- gdm-2.13.0.1.orig/configure	2005-11-14 17:34:33.000000000 -0500
+++ gdm-2.13.0.1/configure	2005-11-18 14:59:31.000000000 -0500
@@ -31700,7 +31700,7 @@
 #define HAVE_SELINUX 1
 _ACEOF
 
-	EXTRA_DAEMON_LIBS="$EXTRA_DAEMON_LIBS -lselinux -lattr"
+	EXTRA_DAEMON_LIBS="$EXTRA_DAEMON_LIBS -lselinux"
 fi
 
 #
diff --exclude-from=/home/dang/.diffrc -u -ruN gdm-2.13.0.1.orig/configure.ac gdm-2.13.0.1/configure.ac
--- gdm-2.13.0.1.orig/configure.ac	2005-11-14 17:13:40.000000000 -0500
+++ gdm-2.13.0.1/configure.ac	2005-11-18 14:59:23.000000000 -0500
@@ -750,7 +750,7 @@
 	AC_CHECK_LIB(selinux,setexeccon,/bin/true)
 	AC_CHECK_LIB(selinux,is_selinux_enabled,/bin/true)
 	AC_DEFINE(HAVE_SELINUX)
-	EXTRA_DAEMON_LIBS="$EXTRA_DAEMON_LIBS -lselinux -lattr"
+	EXTRA_DAEMON_LIBS="$EXTRA_DAEMON_LIBS -lselinux"
 fi
 
 #
