From 8f9bf7b053fc7a6c2e5b33fc43c168ba7250cb98 Mon Sep 17 00:00:00 2001
From: Alexandre Rostovtsev <tetromino@gentoo.org>
Date: Tue, 25 Sep 2012 17:38:37 -0400
Subject: [PATCH] configure: Make selinux check non-automagic

---
 configure.ac | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 80a1fd4..61a43d6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -142,9 +142,13 @@ AC_SUBST(UPOWER)
 AC_SUBST(UPOWER_CFLAGS)
 AC_SUBST(UPOWER_LIBS)
 
+AC_ARG_WITH(selinux,
+	    AS_HELP_STRING([--with-selinux],
+	                   [Add SELinux support]))
+
 PKG_CHECK_MODULES(LIBSELINUX, libselinux, have_selinux=yes, have_selinux=no)
 
-if test "x$have_selinux" = "xyes" ; then
+if test "x$have_selinux" = "xyes" && test "x$with_selinux" != "xno" ; then
         AC_DEFINE(HAVE_SELINUX, 1, [Define if have selinux])
 fi
 AC_SUBST(LIBSELINUX_CFLAGS)
-- 
1.7.12

