From 41badd6d2ca652a875fec3eea420ae876023076d Mon Sep 17 00:00:00 2001
From: Gilles Dartiguelongue <eva@gentoo.org>
Date: Tue, 2 Nov 2010 23:15:54 +0100
Subject: [PATCH 1/6] remove unneeded linker directive for selinux

Gentoo bug: #41022

---
 configure.ac |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/configure.ac b/configure.ac
index ed07dce..71d0247 100644
--- a/configure.ac
+++ b/configure.ac
@@ -873,7 +873,7 @@ if test "x$with_selinux" = "xyes" ; then
 	AC_CHECK_LIB(selinux,setexeccon,/bin/true)
 	AC_CHECK_LIB(selinux,is_selinux_enabled,/bin/true)
 	AC_DEFINE(HAVE_SELINUX, 1, [Define if have selinux])
-	EXTRA_DAEMON_LIBS="$EXTRA_DAEMON_LIBS -lselinux -lattr"
+	EXTRA_DAEMON_LIBS="$EXTRA_DAEMON_LIBS -lselinux"
 fi
 
 dnl ---------------------------------------------------------------------------
-- 
1.7.3.1

