diff -Naur gdm-3.3.92.1//data/00-upstream-settings gdm-3.3.92.1.new//data/00-upstream-settings
--- gdm-3.3.92.1//data/00-upstream-settings	2012-02-08 04:50:32.000000000 +0530
+++ gdm-3.3.92.1.new//data/00-upstream-settings	2012-03-25 14:24:01.387842352 +0530
@@ -7,7 +7,7 @@
 #
 
 [org/gnome/desktop/a11y/keyboard]
-enable=true
+enable=false
 
 [org/gnome/desktop/background]
 show-desktop-icons=false
@@ -16,7 +16,7 @@
 exec='/bin/true'
 
 [org/gnome/desktop/interface]
-toolkit-accessibility=true
+toolkit-accessibility=false
 
 [org/gnome/desktop/lockdown]
 disable-application-handlers=true
@@ -35,7 +35,7 @@
 event-sounds=true
 
 [org/gnome/settings-daemon/plugins/a11y-keyboard]
-active=true
+active=false
 
 [org/gnome/settings-daemon/plugins/a11y-settings]
 active=false
