--- system/dema/session_decoding.c.orig	2012-09-30 19:57:23.000000000 +0800
+++ system/dema/session_decoding.c	2013-09-19 07:59:06.904040050 +0800
@@ -111,7 +111,7 @@
         return 0;
     }
 #ifdef SO_REUSEPORT
-    if (setsockopt(sock, SOL_SOCKET, SO_REUSEPORT,
+    if (setsockopt(sd, SOL_SOCKET, SO_REUSEPORT,
                    (char *) &yes, sizeof(yes)) < 0) {
         perror("SO_REUSEPORT");
         close(sd);
