--- src/phishing/smtp/client/smtp_client.py.orig	2013-01-18 09:43:44.000000000 +0800
+++ src/phishing/smtp/client/smtp_client.py	2013-01-18 09:45:28.630897843 +0800
@@ -29,29 +29,16 @@
     # strip carriage returns
     line=line.rstrip()
     match=re.search("SENDMAIL=",line)
-    if match: 
+    if match:
         # if match and if line is flipped on continue on
         if line == ("SENDMAIL=ON"):
-            print_info("Sendmail is a Linux based SMTP Server, this can be used to spoof email addresses.")
-            print_info("Sendmail can take up to three minutes to start FYI.")
-            print_status("Sendmail is set to ON")
-            sendmail_choice = yesno_prompt(["1"], "Start Sendmail? [yes|no]")
-            # if yes, then do some good stuff
-            if sendmail_choice == "YES":
-                print_info("NOTE: Sendmail can take 3-5 minutes to start.")
-                if os.path.isfile("/etc/init.d/sendmail"):
-                    subprocess.Popen("/etc/init.d/sendmail start", shell=True).wait()
-                # if not there then prompt user
-                if not os.path.isfile("/etc/init.d/sendmail"):
-                    pause=raw_input("[!] Sendmail was not found. Install it and try again.")
-                    sys.exit()
-                smtp = ("localhost")
-                port = ("25")
-                # Flip sendmail switch to get rid of some questions             
-                sendmail=1 
-                # just throw user and password to blank, needed for defining below
-                user=''
-                pwd=''
+            print_info("You need to configure EMAIL_PROVIDER and /etc/ssmtp/* paramters manually before")
+            print_info("running this option")
+            # Flip sendmail switch to get rid of some questions
+            sendmail=1
+            # just throw user and password to blank, needed for defining below
+            user=''
+            pwd=''
 
     # Search for SMTP provider we will be using
     match1=re.search("EMAIL_PROVIDER=", line)
--- src/phishing/smtp/client/smtp_web.py.orig	2013-01-18 09:26:51.000000000 +0800
+++ src/phishing/smtp/client/smtp_web.py	2013-01-18 09:25:42.797840452 +0800
@@ -42,25 +42,13 @@
         if match: 
                 # if match and if line is flipped on continue on
                 if line == ("SENDMAIL=ON"):
-                        print_info("Sendmail is a Linux based SMTP Server, this can be used to spoof email addresses.")
-                        print_info("Sendmail can take up to three minutes to start")
-                        print_status("Sendmail is set to ON")
-                        sendmail_choice = yesno_prompt(["1"], "Start Sendmail? [yes|no]")
-                        # if yes, then do some good stuff
-                        if sendmail_choice == "YES":
-                                print_info("Sendmail can take up to 3-5 minutes to start")
-                                if os.path.isfile("/etc/init.d/sendmail"):
-                                        subprocess.Popen("/etc/init.d/sendmail start", shell=True).wait()
-                                if not os.path.isfile("/etc/init.d/sendmail"):
-                                        pause = raw_input("[!] Sendmail was not found. Try again and restart.")
-                                        sys.exit()
-                                smtp = ("localhost")
-                                port = ("25")
-                                # Flip sendmail switch to get rid of some questions             
-                                sendmail=1 
-                                # just throw user and password to blank, needed for defining below
-                                user=''
-                                pwd=''
+                        print_info("You need to configure EMAIL_PROVIDER and /etc/ssmtp/* paramters manually before")
+                        print_info("running this option")
+                        # Flip sendmail switch to get rid of some questions
+                        sendmail=1
+                        # just throw user and password to blank, needed for defining below
+                        user=''
+                        pwd=''
 
         # Search for SMTP provider we will be using
         match1=re.search("EMAIL_PROVIDER=", line)
