Index: dff/dff/api/gui/video/video.cpp
===================================================================
--- dff.orig/dff/api/gui/video/video.cpp	2013-07-08 13:03:30.372074867 +0200
+++ dff/dff/api/gui/video/video.cpp	2013-08-29 11:28:17.314265471 +0200
@@ -124,7 +124,7 @@
 	this->_clear();	
         throw std::string("can't open input stream");
   }
-  if (av_find_stream_info(this->_formatContext) < 0)
+  if (avformat_find_stream_info(this->_formatContext, NULL) < 0)
   {
 	this->_clear();
 	throw std::string("can't find video info");
@@ -419,7 +419,7 @@
   }
 
   this->_codecContext->workaround_bugs = 1;
-  if (avcodec_open(this->_codecContext, this->_codec) < 0)
+  if (avcodec_open2(this->_codecContext, this->_codec, NULL) < 0)
   {
     this->_clear();
     throw std::string("Could not open video");
