--- dff/ui/gui/mainwindow.py.orig	2011-10-04 23:38:44.000000000 +0800
+++ dff/ui/gui/mainwindow.py	2011-10-10 07:51:41.000000000 +0800
@@ -52,7 +52,11 @@
 from ui.gui.resources.ui_mainwindow import Ui_MainWindow
 
 # Documentation
-from ui.gui.widget.help import Help
+HELP = True
+try:
+    from ui.gui.widget.help import Help
+except ImportError:
+    HELP = False
 
 class MainWindow(QMainWindow, Ui_MainWindow):
     def __init__(self,  app, debug = False):
@@ -99,7 +103,8 @@
         self.connect(self.actionNodeBrowser, SIGNAL("triggered()"), self.addNodeBrowser)
         self.connect(self.actionShell, SIGNAL("triggered()"), self.shellActions.create)
         self.connect(self.actionPython_interpreter, SIGNAL("triggered()"), self.interpreterActions.create)        ## About menu
-        self.connect(self.actionHelp, SIGNAL("triggered()"), self.addHelpWidget)
+        if HELP:
+            self.connect(self.actionHelp, SIGNAL("triggered()"), self.addHelpWidget)
         self.connect(self.actionAbout, SIGNAL("triggered()"), self.dialog.about)
         # list used to build toolbar
         # None will be a separator
@@ -110,12 +115,13 @@
                             self.actionShell,
                             self.actionPython_interpreter,
                             self.actionIdeOpen,
-                            self.actionHelp,
                             None,
                             self.actionMaximize,
                             self.actionFullscreen_mode,
                             self.actionBrowse_modules,
                             ]
+        if HELP:
+            self.toolbarList = [self.actionHelp]
 
         # Set up toolbar
         self.setupToolBar()
