--- beef-0.4.5.0/core/loader.rb.orig	2014-04-25 02:17:17.000000000 +0800
+++ beef-0.4.5.0/core/loader.rb	2014-12-23 15:44:36.000000000 +0800
@@ -6,8 +6,10 @@
 
 # @note Include here all the gems we are using
 require 'rubygems'
-require 'bundler/setup'
-Bundler.require(:default)
+
+#require 'bundler/setup'
+#Bundler.require(:default)
+
 require 'cgi'
 require 'yaml'
 require 'singleton'
@@ -18,6 +20,23 @@
 require 'rubydns'
 require 'sourcify'
 
+require 'eventmachine'
+require 'thin'
+require 'sinatra'
+require 'rack'
+require 'em-websocket'
+require 'jsmin'
+require 'ansi'
+require 'term/ansicolor'
+require 'dm-core'
+require 'json'
+require 'data_objects'
+require 'dm-sqlite-adapter'
+require 'parseconfig'
+require 'erubis'
+require 'dm-migrations'
+require 'msfrpc-client'
+
 # @note Include the filters
 require 'core/filters'
 
