diff -Naur switchproxy_tool-1.4.1-fx+mz+tb.orig/install.rdf switchproxy_tool-1.4.1-fx+mz+tb/install.rdf
--- switchproxy_tool-1.4.1-fx+mz+tb.orig/install.rdf	2009-01-26 15:10:03.000000000 +0000
+++ switchproxy_tool-1.4.1-fx+mz+tb/install.rdf	2009-01-26 15:23:20.000000000 +0000
@@ -29,7 +29,7 @@
 			<Description>
 				<em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
 				<em:minVersion>1.0</em:minVersion>
-				<em:maxVersion>2.0</em:maxVersion>
+				<em:maxVersion>3.1</em:maxVersion>
 			</Description>
 		</em:targetApplication>
 		
