diff -Naur tcpick-0.2.1/src/verify.c tcpick-0.2.1-working/src/verify.c
--- tcpick-0.2.1/src/verify.c	2020-05-05 15:55:06.524010213 -0400
+++ tcpick-0.2.1-working/src/verify.c	2020-05-05 15:56:12.387012864 -0400
@@ -92,7 +92,7 @@
  tcppacket->rst         == 1
 
 
-__inline__ int
+static __inline__ int
 established_packet ( struct CONN * conn_ptr, struct HOST_DESC * Desc)
 /* called by verify()
    packets of established connections come here */
