unchanged:
--- a/spec/request/processing_spec.rb
+++ b/spec/request/processing_spec.rb
@@ -44,6 +44,7 @@ describe Request, 'processing' do
   end
   
   it "should set body external encoding to ASCII_8BIT" do
+    pending("Ruby 1.9 compatible implementations only") unless StringIO.instance_methods.include? :external_encoding
     Request.new.body.external_encoding.should == Encoding::ASCII_8BIT
   end
 end
\ No newline at end of file
only in patch2:
unchanged:
--- a/spec/server_spec.rb
+++ b/spec/server_spec.rb
@@ -12,6 +12,10 @@ describe Server do
   end
 
   it "should set lower maximum_connections size when too large" do
+    # root users under Linux will not have a limitation on maximum
+    # connections, so we cannot really run this test under that
+    # condition.
+    pending("only for non-root users") if Process.euid == 0
     @server.maximum_connections = 100_000
     @server.config
     @server.maximum_connections.should < 100_000
