--- vmware-vix.py.orig	2010-02-04 19:50:56.000000000 +0100
+++ vmware-vix.py	2010-02-05 02:17:54.000000000 +0100
@@ -6,7 +6,8 @@
 DEST = LIBDIR/'vmware-vix'
 SETTINGS = { 'vmware.fullpath': BINDIR/'vmware',
              'vix.libdir': DEST, }
-CONF = DEST/'setup/vmware-config'
+import os
+CONF = path(os.environ['WORKDIR'])/'vmware-config.sh'
 
 class VIX(Installer):
    def PreTransactionInstall(self, old, new, upgrade):
@@ -29,7 +30,7 @@
       for key, val in SETTINGS.items():
          self.RunCommand(CONF, '-s', key, val)
       # Add link to deprecated uninstall mechanism to catch downgrades
-      self.AddUninstallLinks('vix')
+      # self.AddUninstallLinks('vix')
 
    def PreUninstall(self, old, new, upgrade):
       # XXX: VIX may have been removed out from underneath us if
