diff -ur catalyst-2.0.8.1.orig//arch/arm.py catalyst-2.0.8.1/arch/arm.py
--- catalyst-2.0.8.1.orig//arch/arm.py	2012-07-07 20:28:46.788719269 +0200
+++ catalyst-2.0.8.1/arch/arm.py	2012-07-07 20:29:28.099721112 +0200
@@ -67,28 +67,28 @@
 	"Builder class for armv6j target"
 	def __init__(self,myspec):
 		generic_arm.__init__(self,myspec)
-		self.settings["CHOST"]="armv6j-unknown-linux-gnueabi"
+		self.settings["CHOST"]="armv6j-softfp-linux-gnueabi"
 		self.settings["CFLAGS"]+=" -march=armv6j -mfpu=vfp -mfloat-abi=softfp"
 
 class arch_armv6z(generic_arm):
 	"Builder class for armv6z target"
 	def __init__(self,myspec):
 		generic_arm.__init__(self,myspec)
-		self.settings["CHOST"]="armv6z-unknown-linux-gnueabi"
+		self.settings["CHOST"]="armv6z-softfp-linux-gnueabi"
 		self.settings["CFLAGS"]+=" -march=armv6z -mfpu=vfp -mfloat-abi=softfp"
 
 class arch_armv6zk(generic_arm):
 	"Builder class for armv6zk target"
 	def __init__(self,myspec):
 		generic_arm.__init__(self,myspec)
-		self.settings["CHOST"]="armv6zk-unknown-linux-gnueabi"
+		self.settings["CHOST"]="armv6zk-softfp-linux-gnueabi"
 		self.settings["CFLAGS"]+=" -march=armv6zk -mfpu=vfp -mfloat-abi=softfp"
 
 class arch_armv7a(generic_arm):
 	"Builder class for armv7a target"
 	def __init__(self,myspec):
 		generic_arm.__init__(self,myspec)
-		self.settings["CHOST"]="armv7a-unknown-linux-gnueabi"
+		self.settings["CHOST"]="armv7a-softfp-linux-gnueabi"
 		self.settings["CFLAGS"]+=" -march=armv7-a -mfpu=vfpv3-d16 -mfloat-abi=softfp"
 
 class arch_armv6j_hardfp(generic_arm):