diff --git a/targets/support/kmerge.sh b/targets/support/kmerge.sh
index ef3eb2e..880aa1b 100644
--- a/targets/support/kmerge.sh
+++ b/targets/support/kmerge.sh
@@ -202,7 +202,7 @@ fi
 if [ -n "${clst_KERNCACHE}" ]
 then
 	mkdir -p /tmp/kerncache/${clst_kname}
-	clst_root_path=/tmp/kerncache/${clst_kname} PKGDIR=${PKGDIR} clst_myemergeopts="--quiet --update --newuse" run_merge "${clst_ksource}" || exit 1
+	clst_root_path=/tmp/kerncache/${clst_kname} PKGDIR=${PKGDIR} clst_myemergeopts="--quiet --update --newuse --getbinpkg --usepkg --binpkg-respect-use=n" run_merge "${clst_ksource}" || exit 1
 	KERNELVERSION=`portageq best_visible / "${clst_ksource}"`
 	if [ ! -e /etc/portage/profile/package.provided ]
 	then
