diff --git a/modules/generic_stage_target.py b/modules/generic_stage_target.py
index 5947e9a..3a41751 100644
--- a/modules/generic_stage_target.py
+++ b/modules/generic_stage_target.py
@@ -816,9 +816,8 @@ class generic_stage_target(generic_target):
 			cmd("rm -f "+self.settings["chroot_path"]+"/etc/portage/make.profile",\
 					"Error zapping profile link",env=self.env)
 			cmd("mkdir -p "+self.settings["chroot_path"]+"/etc/portage/")
-			cmd("ln -sf ../../usr/portage/profiles/"+\
-				self.settings["target_profile"]+" "+\
-				self.settings["chroot_path"]+"/etc/portage/make.profile",\
+			cmd("eselect profile set "+\
+				self.settings["target_profile"],
 				"Error creating profile link",env=self.env)
 			touch(self.settings["autoresume_path"]+"config_profile_link")
 
diff --git a/targets/stage1/stage1-controller.sh b/targets/stage1/stage1-controller.sh
index 3c348ce..97c66d9 100644
--- a/targets/stage1/stage1-controller.sh
+++ b/targets/stage1/stage1-controller.sh
@@ -10,11 +10,12 @@ case $1 in
 		
 		# Setup "ROOT in chroot" dir
 		install -d ${clst_chroot_path}/${clst_root_path}/etc
+		install -d ${clst_chroot_path}/${clst_root_path}/etc/portage
 		
 		# Setup make.conf and make.profile link in "ROOT in chroot":
-		copy_to_chroot ${clst_chroot_path}/etc/portage/make.conf /${clst_root_path}/etc
+		copy_to_chroot ${clst_chroot_path}/etc/portage/make.conf /${clst_root_path}/etc/portage
 		copy_to_chroot ${clst_chroot_path}/etc/portage/make.profile \
-			/${clst_root_path}/etc
+			/${clst_root_path}/etc/portage
 
 		# Enter chroot, execute our build script
 		exec_in_chroot \