diff --git a/targets/stage1/stage1-controller.sh b/targets/stage1/stage1-controller.sh
index 3c348ce..b481605 100644
--- a/targets/stage1/stage1-controller.sh
+++ b/targets/stage1/stage1-controller.sh
@@ -7,14 +7,14 @@ case $1 in
 	;;
 	run)
 		cp ${clst_sharedir}/targets/stage1/build.py ${clst_chroot_path}/tmp
-		
+
 		# Setup "ROOT in chroot" dir
-		install -d ${clst_chroot_path}/${clst_root_path}/etc
-		
+		install -d ${clst_chroot_path}/${clst_root_path}/etc/portage
+
 		# Setup make.conf and make.profile link in "ROOT in chroot":
-		copy_to_chroot ${clst_chroot_path}/etc/portage/make.conf /${clst_root_path}/etc
+		copy_to_chroot ${clst_chroot_path}/etc/portage/make.conf /${clst_root_path}/etc/portage
 		copy_to_chroot ${clst_chroot_path}/etc/portage/make.profile \
-			/${clst_root_path}/etc
+			/${clst_root_path}/etc/portage
 
 		# Enter chroot, execute our build script
 		exec_in_chroot \
