diff --git a/targets/support/livecdfs-update.sh b/targets/support/livecdfs-update.sh
index fda3e36..a37332d 100644
--- a/targets/support/livecdfs-update.sh
+++ b/targets/support/livecdfs-update.sh
@@ -220,17 +220,6 @@ then
 	fi
 fi
 
-# Create firmware directory if it does not exist
-[ ! -d /lib/firmware ] && mkdir -p /lib/firmware
-
-# tar up the firmware so that it does not get clobbered by the livecd mounts
-if [ -n "$(ls /lib/firmware)" ]
-then
-	cd /lib/firmware
-	/bin/tar -I bzip2 -cpf /lib/firmware.tar.bz2 .
-	rm -rf /lib/firmware/*
-fi
-
 # Clear out locales
 case ${clst_livecd_type} in
 	gentoo-release-minimal|gentoo-release-universal|gentoo-gamecd)
