diff -Naurp nmap-5.20/portlist.cc nmap-5.20-fixed/portlist.cc
--- nmap-5.20/portlist.cc	2009-12-20 04:22:19.000000000 +0100
+++ nmap-5.20-fixed/portlist.cc	2010-01-23 09:59:21.000000000 +0100
@@ -412,7 +412,7 @@ void PortList::setRPCProbeResults(u16 po
   }
 }
 
-
+#ifndef NOLUA
 void PortList::addScriptResult(u16 portno, int protocol, ScriptResult& sr) {
   Port *port;
 
@@ -420,6 +420,7 @@ void PortList::addScriptResult(u16 portn
 
   port->scriptResults.push_back(sr);
 }
+#endif
 
 /*****************************************************************************/
 /* Convert protocol name from in.h to enum portlist_proto.
diff -Naurp nmap-5.20/portlist.h nmap-5.20-fixed/portlist.h
--- nmap-5.20/portlist.h	2009-12-20 04:22:19.000000000 +0100
+++ nmap-5.20-fixed/portlist.h	2010-01-23 09:56:20.000000000 +0100
@@ -278,7 +278,9 @@ class PortList {
   void setRPCProbeResults(u16 portno, int proto, int rpc_status, unsigned long rpc_program, 
 			  unsigned int rpc_lowver, unsigned int rpc_highver);
 
+#ifndef NOLUA
   void addScriptResult(u16 portno, int protocol, ScriptResult& sr);
+#endif
 
   /* Cycles through the 0 or more "ignored" ports which should be
    consolidated for Nmap output.  They are returned sorted by the
