diff -urN nmap-7.90.orig/libnetutil/Makefile.in nmap-7.90/libnetutil/Makefile.in
--- nmap-7.90.orig/libnetutil/Makefile.in	2017-08-23 16:52:40.000000000 +0800
+++ nmap-7.90/libnetutil/Makefile.in	2020-10-08 12:47:12.205110747 +0800
@@ -5,7 +5,7 @@
 CXXFLAGS = @CXXFLAGS@
 CPPFLAGS = @CPPFLAGS@ $(DEFS)
 DEFS = @DEFS@
-DEFS += -D_FORTIFY_SOURCE=2
+DEFS +=
 AR = ar
 RANLIB = @RANLIB@
 
diff -urN nmap-7.90.orig/Makefile.in nmap-7.90/Makefile.in
--- nmap-7.90.orig/Makefile.in	2020-08-28 03:13:31.000000000 +0800
+++ nmap-7.90/Makefile.in	2020-10-08 12:46:48.678109168 +0800
@@ -39,7 +39,7 @@
 # http://gcc.gnu.org/ml/gcc-patches/2004-09/msg02055.html
 # Level 1 only makes changes that don't affect "conforming" programs,
 # while level 2 enforces additional restrictions.
-DEFS += -D_FORTIFY_SOURCE=2
+DEFS +=
 # For mtrace debugging -- see MTRACE define in main.cc for instructions
 # Should only be enabled during debugging and not in any real release.
 # DEFS += -DMTRACE=1
diff -urN nmap-7.90.orig/nbase/Makefile.in nmap-7.90/nbase/Makefile.in
--- nmap-7.90.orig/nbase/Makefile.in	2018-11-01 12:35:00.000000000 +0800
+++ nmap-7.90/nbase/Makefile.in	2020-10-08 12:47:07.150110408 +0800
@@ -12,7 +12,7 @@
 CCOPT = 
 DEFS = @DEFS@
 # With GCC, add extra security checks to source code.
-DEFS += -D_FORTIFY_SOURCE=2
+DEFS +=
 CPPFLAGS = @CPPFLAGS@
 CFLAGS = @CFLAGS@ $(CCOPT) $(GLIB_CFLAGS) $(DEFS) $(INCLS)
 STATIC = 
diff -urN nmap-7.90.orig/ncat/Makefile.in nmap-7.90/ncat/Makefile.in
--- nmap-7.90.orig/ncat/Makefile.in	2018-10-16 22:34:39.000000000 +0800
+++ nmap-7.90/ncat/Makefile.in	2020-10-08 12:47:01.809110049 +0800
@@ -53,7 +53,7 @@
 LIBS = @LIBS@
 DEFS = @DEFS@ -DNCAT_DATADIR="\"$(pkgdatadir)\""
 # With GCC, add extra security checks to source code.
-DEFS += -D_FORTIFY_SOURCE=2
+DEFS +=
 INCLS = -I. -I.. -I../nsock/include/ -I$(NBASEDIR)
 RM = rm -f
 STRIP = @STRIP@
diff -urN nmap-7.90.orig/nping/Makefile.in nmap-7.90/nping/Makefile.in
--- nmap-7.90.orig/nping/Makefile.in	2020-10-02 21:55:49.000000000 +0800
+++ nmap-7.90/nping/Makefile.in	2020-10-08 12:47:30.667111986 +0800
@@ -38,7 +38,7 @@
 # http://gcc.gnu.org/ml/gcc-patches/2004-09/msg02055.html
 # Level 1 only makes changes that don't affect "conforming" programs,
 # while level 2 enforces additional restrictions.
-DEFS += -D_FORTIFY_SOURCE=2
+DEFS +=
 # For mtrace debugging -- see MTRACE define in main.cc for instructions
 # Should only be enabled during debugging and not in any real release.
 # DEFS += -DMTRACE=1
diff -urN nmap-7.90.orig/nsock/src/Makefile.in nmap-7.90/nsock/src/Makefile.in
--- nmap-7.90.orig/nsock/src/Makefile.in	2017-08-23 16:52:40.000000000 +0800
+++ nmap-7.90/nsock/src/Makefile.in	2020-10-08 12:46:57.470109758 +0800
@@ -13,7 +13,7 @@
 CCOPT =
 DEFS = @DEFS@ -DNSOCK_VERSION=\"$(NSOCK_VERSION)\"
 # With GCC, add extra security checks to source code.
-DEFS += -D_FORTIFY_SOURCE=2
+DEFS +=
 INCLS = -I../include
 CFLAGS = @CFLAGS@ $(CCOPT)
 # CFLAGS = -g -Wall $(DEFS) $(INCLS)
