diff -Naurp nmap-4.75-orig/nmap_dns.h nmap-4.75/nmap_dns.h
--- nmap-4.75-orig/nmap_dns.h	2008-09-09 08:51:59.000000000 +0200
+++ nmap-4.75/nmap_dns.h	2008-09-09 08:52:19.000000000 +0200
@@ -95,6 +95,9 @@
  *                                                                         *
  ***************************************************************************/
 
+#include <list>
+#include <string>
+
 class Target;
 
 void nmap_mass_rdns(Target ** targets, int num_targets);
