--- nmap-4.76/zenmap/install_scripts/unix/su-to-zenmap.sh.orig
+++ nmap-4.76/zenmap/install_scripts/unix/su-to-zenmap.sh
@@ -45,7 +45,7 @@
       ktsuss) ktsuss -u "$PRIV" "$COMMAND";;
   # As a last resort, open a new xterm use sudo/su
       sdterm) xterm -e "sudo -u $PRIV $COMMAND";;
-      sterm) xterm -e "su -l $PRIV -C $COMMAND";;
+      sterm) xterm -e "su -l $PRIV -c $COMMAND";;
     esac;
 fi
 
