--- fierce.pl.orig	2008-04-06 18:04:55.000000000 +0000
+++ fierce.pl	2008-12-02 14:10:38.000000000 +0000
@@ -243,10 +243,10 @@
   output("\nUnsuccessful in zone transfer (it was worth a shot)");
 }
   output("Okay, trying the good old fashioned way... brute force");
-  $wordlist = $wordlist || 'hosts.txt';
+  $wordlist = $wordlist || '/usr/share/fierce/hosts.txt';
   if (-e $wordlist) {
     open (WORDLIST, '<', $wordlist)   or
-    open (WORDLIST, '<', 'hosts.txt') or
+    open (WORDLIST, '<', '/usr/share/fierce/hosts.txt') or
     quit_early("Can't open $wordlist or the default wordlist");
     for (<WORDLIST>){
       chomp;
@@ -625,6 +625,8 @@
 	-wordlist	Use a seperate wordlist (one word per line).  Usage:
 
 	perl fierce.pl -dns examplecompany.com -wordlist dictionary.txt
+
+	Default hosts list is in /usr/share/fierce/hosts.txt
 EOHELP
 exit;
 }
