diff -ur a/setup.py b/setup.py
--- a/setup.py	2019-05-09 18:53:05.000000000 +0300
+++ b/setup.py	2020-04-12 17:26:36.746264784 +0300
@@ -49,17 +49,6 @@
     },
     "packages": packages,
     "py_modules": [],
-    "install_requires": [
-        "certifi",
-        "click==7.0",
-        "colorama",
-        "neobolt~=1.7.12",
-        "neotime~=1.7.4",
-        "prompt_toolkit~=2.0.7",
-        "pygments~=2.3.1",
-        "pytz",
-        "urllib3<1.25,>=1.23",
-    ],
     "extras_require": {
     },
     "license": __license__,
