diff -urN libndpi-20140717.orig/src/include/Makefile.am libndpi-20140717/src/include/Makefile.am
--- libndpi-20140717.orig/src/include/Makefile.am	2014-07-19 14:23:00.000000000 +0800
+++ libndpi-20140717/src/include/Makefile.am	2014-07-19 14:24:24.511377042 +0800
@@ -1,9 +1,16 @@
 
-library_includedir=$(includedir)/libndpi-1.4/libndpi
+library_includedir=$(includedir)/libndpi
 
-library_include_HEADERS = ndpi_api.h \
-		  ndpi_debug_functions.h \
-		  ndpi_define.h \
-		  ndpi_macros.h \
-		  ndpi_protocols_osdpi.h \
-		  ndpi_public_functions.h
+library_include_HEADERS = linux_compat.h \
+		ndpi_api.h \
+		ndpi_credis.h \
+		ndpi_debug_functions.h \
+		ndpi_define.h \
+		ndpi_macros.h \
+		ndpi_main.h \
+		ndpi_protocol_history.h \
+		ndpi_protocols.h \
+		ndpi_protocols_osdpi.h \
+		ndpi_public_functions.h \
+		ndpi_structs.h \
+		ndpi_utils.h
diff -urN libndpi-20140717.orig/src/include/Makefile.in libndpi-20140717/src/include/Makefile.in
--- libndpi-20140717.orig/src/include/Makefile.in	2014-07-19 14:23:00.000000000 +0800
+++ libndpi-20140717/src/include/Makefile.in	2014-07-19 14:25:13.471380329 +0800
@@ -189,13 +189,20 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-library_includedir = $(includedir)/libndpi-1.4/libndpi
-library_include_HEADERS = ndpi_api.h \
-		  ndpi_debug_functions.h \
-		  ndpi_define.h \
-		  ndpi_macros.h \
-		  ndpi_protocols_osdpi.h \
-		  ndpi_public_functions.h
+library_includedir = $(includedir)/libndpi
+library_include_HEADERS = linux_compat.h \
+		ndpi_api.h \
+		ndpi_credis.h \
+		ndpi_debug_functions.h \
+		ndpi_define.h \
+		ndpi_macros.h \
+		ndpi_main.h \
+		ndpi_protocol_history.h \
+		ndpi_protocols.h \
+		ndpi_protocols_osdpi.h \
+		ndpi_public_functions.h \
+		ndpi_structs.h \
+		ndpi_utils.h
 
 all: all-am
 
