--- a/cmake/CPM_SQLite3.cmake
+++ b/cmake/CPM_SQLite3.cmake
@@ -1,11 +1,2 @@
-CPMAddPackage(
-    NAME sqlite3
-    URL https://sqlite.org/2026/sqlite-amalgamation-3530100.zip
-    URL_HASH SHA3_256=3c07136e4f6b5dd0c395be86455014039597bc65b6851f7111e88f71b6e06114
-)
-
-if(sqlite3_ADDED)
-    add_library(sqlite3 STATIC "${sqlite3_SOURCE_DIR}/sqlite3.c")
-    target_include_directories(sqlite3 PUBLIC "${sqlite3_SOURCE_DIR}")
-    set_target_properties(sqlite3 PROPERTIES POSITION_INDEPENDENT_CODE ON)
-endif()
+find_package(SQLite3 REQUIRED)
+add_library(sqlite3 ALIAS SQLite3::SQLite3)
