diff -Naur linux-2.6.28-pentoo-r1-orig/drivers/net/wireless/ath5k/base.c linux-2.6.28-pentoo-r1-improved/drivers/net/wireless/ath5k/base.c
--- linux-2.6.28-pentoo-r1-orig/drivers/net/wireless/ath5k/base.c	2009-01-18 17:49:00.000000000 -0500
+++ linux-2.6.28-pentoo-r1-improved/drivers/net/wireless/ath5k/base.c	2009-01-18 17:56:09.000000000 -0500
@@ -1175,7 +1175,9 @@
 	bf->skbaddr = pci_map_single(sc->pdev, skb->data, skb->len,
 			PCI_DMA_TODEVICE);
 
-	if (info->flags & IEEE80211_TX_CTL_NO_ACK)
+	if (info->flags & IEEE80211_TX_CTL_NO_ACK ||
+	   (info->flags & IEEE80211_TX_CTL_INJECTED &&
+	   !(ieee80211_has_morefrags(((struct ieee80211_hdr *)skb->data)->frame_control))))
 		flags |= AR5K_TXDESC_NOACK;
 
 	pktlen = skb->len;
