diff -Naur compat-wireless-2.6.36-rc5-2-s/config.mk compat-wireless-2.6.36-rc5-2-s-untypoed/config.mk
--- compat-wireless-2.6.36-rc5-2-s/config.mk	2010-09-27 19:54:28.000000000 -0400
+++ compat-wireless-2.6.36-rc5-2-s-untypoed/config.mk	2010-10-02 22:17:31.000000000 -0400
@@ -501,7 +501,7 @@
 CONFIG_RT2X00_LIB_LEDS=y
 endif #CONFIG_COMPAT_KERNEL_25
 # CONFIG_RT2X00_DEBUG=y
-# CONFIG_RT2X00_LIB_DEBUGFS
+# CONFIG_RT2X00_LIB_DEBUGFS=y
 endif
 
 ifeq ($(NEED_RT2X00_FIRMWARE),y)
