diff -Naur compat-wireless-2.6.30-rc6.orig/Makefile compat-wireless-2.6.30-rc6/Makefile
--- compat-wireless-2.6.30-rc6.orig/Makefile	2009-05-17 21:48:36.000000000 +0000
+++ compat-wireless-2.6.30-rc6/Makefile	2009-05-27 12:21:51.000000000 +0000
@@ -3,13 +3,10 @@
 ifneq ($(origin $(KLIB)), undefined)
 KMODPATH_ARG:=  "INSTALL_MOD_PATH=$(KLIB)"
 else
-export KLIB:=          /lib/modules/$(shell uname -r)
+export KLIB:=          $(DESTDIR)/lib/modules/$(KVER)
 endif
 export KLIB_BUILD ?=	$(KLIB)/build
 # Sometimes not available in the path
-MODPROBE := /sbin/modprobe
-MADWIFI=$(shell $(MODPROBE) -l ath_pci)
-OLD_IWL=$(shell $(MODPROBE) -l iwl4965)
 
 ifneq ($(KERNELRELEASE),)
 
@@ -59,92 +56,26 @@
 	@md5sum $(COMPAT_CONFIG) > $(CONFIG_CHECK)
 
 install: uninstall modules
-	$(MAKE) -C $(KLIB_BUILD) M=$(PWD) $(KMODDIR_ARG) $(KMODPATH_ARG) \
+	$(MAKE) -C /lib/modules/$(KVER)/build M=$(PWD) $(KMODDIR_ARG) $(KMODPATH_ARG) \
 		modules_install
 	@# All the scripts we can use
-	@mkdir -p /usr/lib/compat-wireless/
-	@install scripts/modlib.sh	/usr/lib/compat-wireless/
-	@install scripts/madwifi-unload	/usr/sbin/
+	@mkdir -p $(DESTDIR)/usr/lib/compat-wireless/
+	@install scripts/modlib.sh	$(DESTDIR)/usr/lib/compat-wireless/
+	@mkdir -p $(DESTDIR)/usr/sbin/
+	@install scripts/madwifi-unload	$(DESTDIR)/usr/sbin/
 	@# This is to allow switching between drivers without blacklisting
-	@install scripts/athenable	/usr/sbin/
-	@install scripts/b43enable	/usr/sbin/
-	@install scripts/iwl-enable	/usr/sbin/
-	@install scripts/athload	/usr/sbin/
-	@install scripts/b43load	/usr/sbin/
-	@install scripts/iwl-load	/usr/sbin/
-	@if [ ! -z $(MADWIFI) ]; then \
-		echo ;\
-		echo -n "Note: madwifi detected, we're going to disable it. "  ;\
-		echo "If you would like to enable it later you can run:"  ;\
-		echo "    sudo athenable madwifi"  ;\
-		echo ;\
-		echo Running athenable ath5k...;\
-		/usr/sbin/athenable ath5k ;\
-	fi
-	@if [ ! -z $(OLD_IWL) ]; then \
-		echo ;\
-		echo -n "Note: iwl4965 detected, we're going to disable it. "  ;\
-		echo "If you would like to enable it later you can run:"  ;\
-		echo "    sudo iwl-load iwl4965"  ;\
-		echo ;\
-		echo Running iwl-enable iwlagn...;\
-		/usr/sbin/iwl-enable iwlagn ;\
-	fi
+	@install scripts/athenable	$(DESTDIR)/usr/sbin/
+	@install scripts/b43enable	$(DESTDIR)/usr/sbin/
+	@install scripts/iwl-enable	$(DESTDIR)/usr/sbin/
+	@install scripts/athload	$(DESTDIR)/usr/sbin/
+	@install scripts/b43load	$(DESTDIR)/usr/sbin/
+	@install scripts/iwl-load	$(DESTDIR)/usr/sbin/
 	@# If on distributions like Mandriva which like to
 	@# compress their modules this will find out and do
 	@# it for you. Reason is some old version of modutils
 	@# won't know mac80211.ko should be used instead of
 	@# mac80211.ko.gz
 	@./scripts/compress_modules
-	@/sbin/depmod -ae
-	@echo
-	@echo "Currently detected wireless subsystem modules:"
-	@echo 
-	@$(MODPROBE) -l mac80211
-	@$(MODPROBE) -l cfg80211
-	@$(MODPROBE) -l lib80211
-	@$(MODPROBE) -l adm8211
-	@$(MODPROBE) -l at76c50x-usb
-	@$(MODPROBE) -l ath5k
-	@$(MODPROBE) -l ath9k
-	@$(MODPROBE) -l b43
-	@$(MODPROBE) -l b43legacy
-	@$(MODPROBE) -l b44
-	@$(MODPROBE) -l ssb
-	@$(MODPROBE) -l rc80211_simple
-	@$(MODPROBE) -l iwlcore
-	@$(MODPROBE) -l iwl3945
-	@$(MODPROBE) -l iwlagn
-	@$(MODPROBE) -l ipw2100
-	@$(MODPROBE) -l ipw2200
-	@$(MODPROBE) -l libipw
-	@$(MODPROBE) -l lib80211
-	@$(MODPROBE) -l lib80211_crypt
-	@$(MODPROBE) -l libertas_cs
-	@$(MODPROBE) -l libertas_tf
-	@$(MODPROBE) -l libertas_tf_usb
-	@$(MODPROBE) -l ub8xxx
-	@$(MODPROBE) -l p54pci
-	@$(MODPROBE) -l p54usb
-	@$(MODPROBE) -l rt2400pci
-	@$(MODPROBE) -l rt2500pci
-	@$(MODPROBE) -l rt2500usb
-	@$(MODPROBE) -l rt61pci
-	@$(MODPROBE) -l rt73usb
-	@$(MODPROBE) -l usbnet
-	@$(MODPROBE) -l cdc_ether
-	@$(MODPROBE) -l rndis_host
-	@$(MODPROBE) -l rndis_wlan
-	@$(MODPROBE) -l rtl8180
-	@$(MODPROBE) -l rtl8187
-	@$(MODPROBE) -l zd1211rw
-	@echo 
-	@echo Now run:
-	@echo 
-	@echo make unload
-	@echo
-	@echo And then load the wireless module you need. If unsure reboot.
-	@echo
 
 uninstall:
 	@# New location, matches upstream
