diff -Naur genkernel-orig/defaults/linuxrc genkernel/defaults/linuxrc
--- genkernel-orig/defaults/linuxrc	2013-07-02 17:48:07.496212000 -0400
+++ genkernel/defaults/linuxrc	2013-07-02 17:48:57.464131000 -0400
@@ -956,11 +956,9 @@
 		chmod 755 ${CHROOT}/$i
 	done
 	# This will prevent from putting junk on the CHANGESDEV
-	mkdir -p ${CHROOT}/usr/portage/distfiles
 	mount -t tmpfs tmpfs ${CHROOT}/var/tmp
 	mount -t tmpfs tmpfs ${CHROOT}/tmp
-	mount -t tmpfs tmpfs ${CHROOT}/usr/portage/distfiles
-	warn_msg "/tmp /var/tmp /usr/portage/distfiles are mounted in ram"
+	warn_msg "/tmp /var/tmp are mounted in ram"
 	warn_msg "consider saving important files elsewhere..."
 	read -t 3 UNUSEDVAL
 	mount -o bind ${NEW_ROOT}/mnt/cdrom ${CHROOT}/mnt/cdrom
