--- CMakeLists.txt.orig	2011-04-15 21:30:03.000000000 +0800
+++ CMakeLists.txt	2011-04-25 16:38:08.882000039 +0800
@@ -257,7 +257,7 @@
 
 ## Install
 
-install (CODE "file (MAKE_DIRECTORY ${OPENVAS_STATE_DIR}/openvasmd/report_formats/)")
+install (CODE "file (MAKE_DIRECTORY \$ENV{DESTDIR}${OPENVAS_STATE_DIR}/openvasmd/report_formats)")
 
 install (FILES ${CMAKE_BINARY_DIR}/src/openvasmd_log.conf
          DESTINATION ${OPENVAS_SYSCONF_DIR})
