diff -uNr ettercap-NG-0.7.3.orig/src/Makefile.am ettercap-NG-0.7.3/src/Makefile.am
--- ettercap-NG-0.7.3.orig/src/Makefile.am	2006-04-03 19:33:36.000000000 +0100
+++ ettercap-NG-0.7.3/src/Makefile.am	2006-04-03 19:35:09.000000000 +0100
@@ -140,7 +140,9 @@
 ettercap_LDADD += interfaces/gtk/libec_gtk.a
 endif
 
-ettercap_LDFLAGS = -export-dynamic @EC_LIBS@ 
+ettercap_LDADD += @EC_LIBS@
+
+ettercap_LDFLAGS = -export-dynamic
 
 CLEANFILES = *.log *.out core* missing/*.o *~
 
