--- src/util.c.orig	2012-12-26 19:25:40.109336640 +0800
+++ src/util.c	2012-12-27 05:30:31.140189964 +0800
@@ -16,7 +16,8 @@
  * @author Konrad Rieck (konrad.rieck@tu-berlin.de)
  * @{
  */
- 
+
+#include <sys/types.h>
 #include <archive.h>
 #include <archive_entry.h>
 
--- src/farray.c.orig	2012-12-27 05:30:04.309188584 +0800
+++ src/farray.c	2012-12-27 05:30:24.708189605 +0800
@@ -22,6 +22,7 @@
  * @{
  */
 
+#include <sys/types.h>
 #include <archive.h>
 #include <archive_entry.h>
 
