--- wifite.py.org	2014-09-18 18:13:56.000000000 +0800
+++ wifite.py	2014-09-18 18:14:56.642235807 +0800
@@ -437,7 +437,7 @@
                         print R + ' [!]' + O + ' file not found: ' + R + capfile + '\n' + W
                         self.exit_gracefully(1)
             if options.update:
-                self.upgrade()
+                print 'The direct upgrade of this binary has been disabled. Please use a package manager instead'
                 exit(0)
             if options.cracked:
                 if len(self.CRACKED_TARGETS) == 0:
