reverted:
--- b/gplaycli/gplaycli.py
+++ a/gplaycli/gplaycli.py
@@ -32,7 +32,7 @@
 from gpapi.googleplay import RequestError
 from google.protobuf.message import DecodeError
 from pkg_resources import get_distribution, DistributionNotFound
+from pyaxmlparser import APK
-from androguard.core.bytecodes.apk import APK
 
 from clint.textui import progress
 
@@ -494,7 +494,7 @@
 			apk = APK(filepath)
 			packagename = apk.package
 			package_bunch.append(packagename)
+			version_codes.append(util.vcode(apk.version_code))
-			version_codes.append(util.vcode(apk.get_androidversion_code()))
 
 		# BulkDetails requires only one HTTP request
 		# Get APK info from store
reverted:
--- b/requirements.txt
+++ a/requirements.txt
@@ -1,4 +1,4 @@
 protobuf
 gpapi
+pyaxmlparser
-androguard
 clint
reverted:
--- b/setup.py
+++ a/setup.py
@@ -33,7 +33,7 @@
 		install_requires=[
 				'protobuf',
 				'gpapi == 0.4.2',
+				'pyaxmlparser',
-				'androguard',
 				'clint',
 		],
 )
