Index: cpyrit_opencl/setup.py
===================================================================
--- cpyrit_opencl/setup.py	(revision 294)
+++ cpyrit_opencl/setup.py	(working copy)
@@ -52,7 +52,7 @@
         pass
     for path in ('/usr/local/opencl/OpenCL/common/inc', \
                 '/opt/opencl/OpenCL/common/inc', \
-                '/usr/local/opencl/include', \
+                '/opt/ati-stream-sdk/include', \
                 '/usr/local/cuda/include'):
         if os.path.exists(path):
             OPENCL_INC_DIRS.append(path)
