diff -ur a/tools/validation_runner/buildbot.cfg b/tools/validation_runner/buildbot.cfg
--- a/tools/validation_runner/buildbot.cfg	2019-09-10 14:05:06.000000000 +0300
+++ b/tools/validation_runner/buildbot.cfg	2019-10-21 11:25:25.317577537 +0300
@@ -1,5 +1,5 @@
 threads: 1
-validator_path: /home/buildbot/buildbot_slave/qbdi-linux-X86_64/build/QBDI/build/tools/validator/libvalidator2.so
+validator_path: /usr/lib64/libvalidator.so
 database: validation.db
 tests:
 # Filesystem
diff -ur a/tools/validation_runner/coverage.cfg b/tools/validation_runner/coverage.cfg
--- a/tools/validation_runner/coverage.cfg	2019-09-10 14:05:06.000000000 +0300
+++ b/tools/validation_runner/coverage.cfg	2019-10-21 11:25:39.045546901 +0300
@@ -1,5 +1,5 @@
 threads: 4
-validator_path: ../../build/tools/validator/libvalidator.so
+validator_path: /usr/lib64/libvalidator.so
 database: test.db
 tests:
 # Filesystem
diff -ur a/tools/validation_runner/ValidationRunner.py b/tools/validation_runner/ValidationRunner.py
--- a/tools/validation_runner/ValidationRunner.py	2019-09-10 14:05:06.000000000 +0300
+++ b/tools/validation_runner/ValidationRunner.py	2019-10-21 11:33:52.025339764 +0300
@@ -1,3 +1,4 @@
+#!/usr/bin/python
 #
 # This file is part of QBDI.
 #
