From f434c902a521deb3e6d057c3ffa4c11f9875a81b Mon Sep 17 00:00:00 2001
From: Daniel Mierswa <impulze@impulze.org>
Date: Wed, 30 Dec 2009 21:02:08 +0100
Subject: [PATCH] Use IFACE variable instead of iface.

Signed-off-by: Daniel Mierswa <impulze@impulze.org>
---
 net/vlan.sh |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/net/vlan.sh b/net/vlan.sh
index 48838ef..d54adeb 100644
--- a/net/vlan.sh
+++ b/net/vlan.sh
@@ -87,7 +87,7 @@ vlan_post_start()
 
 		# We need to work out the interface name of our new vlan id
 		local ifname="$(sed -n -e \
-			's/^\([^[:space:]]*\) *| '"${vlan}"' *| .*'"${iface}"'$/\1/p' \
+			's/^\([^[:space:]]*\) *| '"${vlan}"' *| .*'"${IFACE}"'$/\1/p' \
 			/proc/net/vlan/config )"
 		mark_service_started "net.${ifname}"
 		(
-- 
1.6.4.4

