From bb45f0eabc8e768296cdf678ff8552a28342080a Mon Sep 17 00:00:00 2001
From: Roy Marples <roy@marples.name>
Date: Fri, 8 Jan 2010 00:22:04 +0000
Subject: [PATCH] Fix BSD compile.

---
 conf.d/Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/conf.d/Makefile b/conf.d/Makefile
index b985271..034dc74 100644
--- a/conf.d/Makefile
+++ b/conf.d/Makefile
@@ -11,7 +11,7 @@ include ${MK}/scripts.mk
 
 SOS?=	BSD
 
-network: network.in network.${OS}
+network: network.in network.${SOS}
 	cp $@.in $@
 	[ -e $@.${SOS} ] && cat $@.${SOS} >> $@ || true
 
-- 
1.6.4.4

