From aa6d81180167192ad41a73e896f58db89a81d371 Mon Sep 17 00:00:00 2001
From: William Hubbs <w.d.hubbs@gmail.com>
Date: Mon, 19 Oct 2009 22:25:40 -0500
Subject: [PATCH] fix iproute2 support for gentoo bug 289762

---
 init.d/network.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/init.d/network.in b/init.d/network.in
index 5e60963..396c67d 100644
--- a/init.d/network.in
+++ b/init.d/network.in
@@ -109,11 +109,11 @@ runip()
 	local int="$1" err=
 
 	shift
-	err=$(LC_ALL=C ip address add "$@" dev "$int" 2>&1)
+	err=$(LC_ALL=C ip address add "$@" dev "$int" brd + 2>&1)
 	if [ -z "$err" ]; then
 		# ip does not bring up the interface when adding addresses
-		if ! intup; then
-			ip set link up dev "$int"
+		if ! $intup; then
+			ip link set "$int" up
 			intup=true
 		fi
 		return 0
-- 
1.6.4.4

