From 4a892a23ba440a90d8a55f6d19d686917fd3e76e Mon Sep 17 00:00:00 2001
From: Nao Nakashima <nao.nakashima@gmail.com>
Date: Wed, 4 Nov 2009 19:40:52 +0000
Subject: [PATCH 3/3] Fix ppp options ordering.
 Fixes #205.

---
 net/pppd.sh |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/net/pppd.sh b/net/pppd.sh
index 5967f0b..a8c8882 100644
--- a/net/pppd.sh
+++ b/net/pppd.sh
@@ -90,7 +90,7 @@ pppd_pre_start()
 	eval passwordset=\$\{password_${IFVAR}-x\}
 	if [ -n "${username}" ] \
 	&& [ -n "${password}" -o -z "${passwordset}" ]; then
-		opts="${opts} plugin passwordfd.so passwordfd 0"
+		opts="plugin passwordfd.so ${opts} passwordfd 0"
 	fi
 	
 	if ! ${hasdefaultmetric}; then
@@ -155,7 +155,7 @@ pppd_pre_start()
 			opts="${opts} connect true"
 			set -- "$@" "${link}"
 		fi
-		opts="${opts} plugin $1.so"
+		opts="plugin $1.so ${opts}"
 		shift
 		opts="${opts} $@"
 	done
-- 
1.6.4.4

