--- rtl8180/r8180_wx.c~	2005-05-08 12:27:05.000000000 +0200
+++ rtl8180/r8180_wx.c	2005-05-13 17:23:59.147295921 +0200
@@ -968,7 +968,7 @@
 #else
 
 struct iw_handler_def  r8180_wx_handlers_def={
-	.standard r8180_wx_handlers,
+	.standard = r8180_wx_handlers,
 	.num_standard = sizeof(r8180_wx_handlers) / sizeof(iw_handler),
 	.private = r8180_private_handler,
 	.num_private = sizeof(r8180_private_handler) / sizeof(iw_handler),
