diff -ur a/xsstrike.py b/xsstrike.py
--- a/xsstrike.py	2019-04-08 11:26:10.000000000 +0300
+++ b/xsstrike.py	2019-12-10 09:44:33.284473315 +0300
@@ -40,8 +40,6 @@
 parser.add_argument('-e', '--encode', help='encode payloads', dest='encode')
 parser.add_argument('--fuzzer', help='fuzzer',
                     dest='fuzz', action='store_true')
-parser.add_argument('--update', help='update',
-                    dest='update', action='store_true')
 parser.add_argument('--timeout', help='timeout',
                     dest='timeout', type=int, default=core.config.timeout)
 parser.add_argument('--proxy', help='use prox(y|ies)',
