--- setup.py.orig	2009-11-21 14:30:29.756444115 +0100
+++ setup.py	2009-11-21 14:33:18.623230521 +0100
@@ -27,20 +27,9 @@
 import subprocess
 
 # Try to find the Brook+ library and headers
-STREAM_LIB_DIRS = []
-STREAM_INC_DIRS = []
+STREAM_LIB_DIRS = ['/opt/atibrook/include','/opt/atibrook/include/brook']
+STREAM_INC_DIRS = ['/opt/atibrook/lib','/usr/lib']
 BRCC = 'brcc'
-for path in ('/usr/local/atibrook/sdk', '/opt/atibrook/sdk'):
-    if os.path.exists(path):
-        STREAM_LIB_DIRS.append(os.path.sep.join((path, 'lib')))
-        STREAM_INC_DIRS.append(os.path.sep.join((path, 'include')))
-        BRCC = os.path.sep.join((path, 'bin', 'brcc'))
-        break
-else:
-    print >>sys.stderr, "The AMD-Stream compiler, headers and libraries " \
-                        "required to build the kernel were not found. " \
-                        "Trying to continue anyway..."
-
 
 try:
     svn_info = subprocess.Popen(('svn', 'info'), \
