Index: cpyrit_stream/setup.py
===================================================================
--- cpyrit_stream/setup.py	(revision 168)
+++ cpyrit_stream/setup.py	(working copy)
@@ -27,19 +27,10 @@
 import subprocess
 
 # Try to find the Brook+ library and headers
-STREAM_LIB_DIRS = []
-STREAM_INC_DIRS = []
+STREAM_INC_DIRS = ['/opt/atibrook/include','/opt/atibrook/include/brook']
+STREAM_LIB_DIRS = ['/opt/atibrook/lib','/usr/lib']
 BRCC = 'brcc'
-for path in ('/usr/local/atibrook/sdk','/opt/atibrook/sdk'):
-    if os.path.exists(path):
-        STREAM_LIB_DIRS.append(os.path.sep.join((path, 'lib')))
-        STREAM_INC_DIRS.append(os.path.sep.join((path, 'include')))
-        BRCC = os.path.sep.join((path, 'bin', 'brcc'))
-        break
-else:
-    print >>sys.stderr, "The AMD-Stream compiler, headers and libraries required to build the kernel were not found. Trying to continue anyway..."
 
-
 try:
     svn_info = subprocess.Popen(('svn', 'info'), stdout=subprocess.PIPE).stdout.read()
     version_string = '0.2.4 (svn r%i)' % int(re.compile('Revision: ([0-9]*)').findall(svn_info)[0])
