Index: cpyrit_stream/setup.py
===================================================================
--- cpyrit_stream/setup.py	(revision 89)
+++ cpyrit_stream/setup.py	(working copy)
@@ -29,19 +29,10 @@
 INCLUDE_DIRS = []
 
 # Try to find the Brook+ library and headers
-STREAM_LIB_DIRS = []
-STREAM_INC_DIRS = []
+STREAM_INC_DIRS = ['/opt/atibrook/include','/opt/atibrook/include/brook']
+STREAM_LIB_DIRS = ['/opt/atibrook/lib','/usr/lib']
 BRCC = 'brcc'
-for path in ('/usr/local/atibrook/sdk','/opt/atibrook/sdk'):
-    if os.path.exists(path):
-        STREAM_LIB_DIRS.append(os.path.sep.join((path, 'lib')))
-        STREAM_INC_DIRS.append(os.path.sep.join((path, 'include')))
-        BRCC = os.path.sep.join((path, 'bin', 'brcc'))
-        break
-else:
-    print >>sys.stderr, "The AMD-Stream compiler, headers and libraries required to build the kernel were not found. Trying to continue anyway..."
 
-
 # Custom build_ext phase to create the GPU code with special compilers before building the whole thing
 class GPUBuilder(build_ext):
     def _call(self, comm):
