diff -ur xfce4-session-4.7.1.orig/xfce4-session/xfsm-shutdown-helper.c xfce4-session-4.7.1/xfce4-session/xfsm-shutdown-helper.c
--- xfce4-session-4.7.1.orig/xfce4-session/xfsm-shutdown-helper.c	2010-11-03 22:11:37.000000000 +0200
+++ xfce4-session-4.7.1/xfce4-session/xfsm-shutdown-helper.c	2010-11-12 20:22:22.000000000 +0200
@@ -335,7 +335,7 @@
   const gchar *consolekit_cookie;
 #endif
   GValue hash_elem = { 0 };
-  gboolean subject_created = FALSE;
+//  gboolean subject_created = FALSE;
 
   helper->polkit_proxy =
     dbus_g_proxy_new_for_name (helper->system_bus,
@@ -394,7 +394,7 @@
               g_hash_table_insert (helper->polkit_subject_hash, g_strdup ("session-id"), &val);
 
               g_free (consolekit_session);
-              subject_created = TRUE;
+              //subject_created = TRUE;
             }
           else if (error)
             {
@@ -410,7 +410,7 @@
    * We failed to get valid session data, then we try
    * to check authentication using the pid.
    **/
-  if (subject_created == FALSE)
+  //if (subject_created == FALSE)
     {
       gint pid;
       guint64 start_time;
