Index: scripts/airdrop-ng/airdrop/libOuiParse.py
===================================================================
--- scripts/airdrop-ng/airdrop/libOuiParse.py	(revision 2109)
+++ scripts/airdrop-ng/airdrop/libOuiParse.py	(working copy)
@@ -34,7 +34,7 @@
 		generate the two dictionaries and return them
 		"""
 		#a poor fix where if we have no file it trys to download it
-		aircrackOUI = '/usr/local/etc/aircrack-ng/airodump-ng-oui.txt'
+		aircrackOUI = '/etc/aircrack-ng/airodump-ng-oui.txt'
 		self.ouiTxtUrl   = "http://standards.ieee.org/regauth/oui/oui.txt"
 		self.ouiUnPath   = install_dir#path to oui.txt if module is installed
 		self.ouiInPath   = install_dir + '/support/'         #path to oui.txt if module is not installed
