diff -Naur aircrack-ng-1.1/scripts/airodump-ng-oui-update aircrack-ng-1.1-patched/scripts/airodump-ng-oui-update
--- aircrack-ng-1.1/scripts/airodump-ng-oui-update	2009-08-07 16:53:15.000000000 -0400
+++ aircrack-ng-1.1-patched/scripts/airodump-ng-oui-update	2011-07-06 10:35:18.000000000 -0400
@@ -4,7 +4,7 @@
 WGET=`which wget 2>/dev/null`
 OUI_DOWNLOAD_URL="http://standards.ieee.org/regauth/oui/oui.txt"
 
-OUI_PATH="/usr/local/etc/aircrack-ng"
+OUI_PATH="/etc/aircrack-ng"
 AIRODUMP_NG_OUI="${OUI_PATH}/airodump-ng-oui.txt"
 OUI_IEEE="${OUI_PATH}/oui.txt"
 USERID=""
