Index: src/airodump-ng.c
===================================================================
--- src/airodump-ng.c	(revision 1916)
+++ src/airodump-ng.c	(working copy)
@@ -589,42 +589,44 @@
 "  usage: airodump-ng <options> <interface>[,<interface>,...]\n"
 "\n"
 "  Options:\n"
-"      --ivs               : Save only captured IVs\n"
-"      --gpsd              : Use GPSd\n"
-"      --write    <prefix> : Dump file prefix\n"
-"      -w                  : same as --write \n"
-"      --beacons           : Record all beacons in dump file\n"
-"      --update     <secs> : Display update delay in seconds\n"
-"      --showack           : Prints ack/cts/rts statistics\n"
-"      -h                  : Hides known stations for --showack\n"
-"      -f          <msecs> : Time in ms between hopping channels\n"
-"      --berlin     <secs> : Time before removing the AP/client\n"
-"                            from the screen when no more packets\n"
-"                            are received (Default: 120 seconds)\n"
-"      -r           <file> : Read packets from that file\n"
-"      -x          <msecs> : Active Scanning Simulation\n"
+"      --ivs                 : Save only captured IVs\n"
+"      --gpsd                : Use GPSd\n"
+"      --write      <prefix> : Dump file prefix\n"
+"      -w                    : same as --write \n"
+"      --beacons             : Record all beacons in dump file\n"
+"      --update       <secs> : Display update delay in seconds\n"
+"      --showack             : Prints ack/cts/rts statistics\n"
+"      -h                    : Hides known stations for --showack\n"
+"      -f            <msecs> : Time in ms between hopping channels\n"
+"      --berlin       <secs> : Time before removing the AP/client\n"
+"                              from the screen when no more packets\n"
+"                              are received (Default: 120 seconds)\n"
+"      -r             <file> : Read packets from that file\n"
+"      -x            <msecs> : Active Scanning Simulation\n"
 "      --output-format\n"
-"                <formats> : Output format. Possible values:\n"
-"                            pcap, ivs, csv, gps, kismet, netxml\n"
+"                  <formats> : Output format. Possible values:\n"
+"                              pcap, ivs, csv, gps, kismet, netxml\n"
+"      --ignore-negative-one : Removes the message that says\n"
+"                              fixed channel <interface>: -1\n"
 "\n"
 "  Filter options:\n"
-"      --encrypt   <suite> : Filter APs by cipher suite\n"
-"      --netmask <netmask> : Filter APs by mask\n"
-"      --bssid     <bssid> : Filter APs by BSSID\n"
-"      -a                  : Filter unassociated clients\n"
+"      --encrypt   <suite>   : Filter APs by cipher suite\n"
+"      --netmask <netmask>   : Filter APs by mask\n"
+"      --bssid     <bssid>   : Filter APs by BSSID\n"
+"      -a                    : Filter unassociated clients\n"
 "\n"
 "  By default, airodump-ng hop on 2.4GHz channels.\n"
 "  You can make it capture on other/specific channel(s) by using:\n"
-"      --channel <channels>: Capture on specific channels\n"
-"      --band <abg>        : Band on which airodump-ng should hop\n"
-"      -C    <frequencies> : Uses these frequencies in MHz to hop\n"
-"      --cswitch  <method> : Set channel switching method\n"
-"                    0     : FIFO (default)\n"
-"                    1     : Round Robin\n"
-"                    2     : Hop on last\n"
-"      -s                  : same as --cswitch\n"
+"      --channel <channels>  : Capture on specific channels\n"
+"      --band <abg>          : Band on which airodump-ng should hop\n"
+"      -C    <frequencies>   : Uses these frequencies in MHz to hop\n"
+"      --cswitch  <method>   : Set channel switching method\n"
+"                    0       : FIFO (default)\n"
+"                    1       : Round Robin\n"
+"                    2       : Hop on last\n"
+"      -s                    : same as --cswitch\n"
 "\n"
-"      --help              : Displays this usage screen\n"
+"      --help                : Displays this usage screen\n"
 "\n";
 
 int is_filtered_netmask(uchar *bssid)
@@ -5037,6 +5039,7 @@
     for(i=0; i<cards; i++)
     {
         chan = wi_get_channel(wi[i]);
+        if(G.ignore_negative_one == 1 && chan==-1) return 0;
         if(G.channel[i] != chan)
         {
             memset(G.message, '\x00', sizeof(G.message));
@@ -5237,6 +5240,7 @@
         {"showack",  0, 0, 'A'},
         {"detect-anomaly", 0, 0, 'E'},
         {"output-format",  1, 0, 'o'},
+        {"ignore-negative-one", 0, &G.ignore_negative_one, 1},
         {0,          0, 0,  0 }
     };
 
Index: src/airodump-ng.h
===================================================================
--- src/airodump-ng.h	(revision 1916)
+++ src/airodump-ng.h	(working copy)
@@ -418,6 +418,8 @@
     pthread_mutex_t mx_sort;			 /* lock write access to ap LL   */
     
     uchar selected_bssid[6];	/* bssid that is selected */
+
+    int ignore_negative_one;
 }
 G;
 
Index: manpages/airodump-ng.1
===================================================================
--- manpages/airodump-ng.1	(revision 1916)
+++ manpages/airodump-ng.1	(working copy)
@@ -58,6 +58,9 @@
 'pcap' is for recording a capture in pcap format, 'ivs' is for ivs format (it is a shortcut for --ivs). 'csv' will create an airodump-ng CSV file, 'kismet' will create a kismet csv file and 'kismet-newcore' will create the kismet netxml file. 'gps' is a shortcut for --gps.
 .br
 Theses values can be combined with the exception of ivs and pcap.
+.TP
+.I --ignore-negative-one
+Removes the message that says 'fixed channel <interface>: -1'.
 .PP
 .B Filter options:
 .TP
