diff -ur aircrack-ng-0.6/src/airodump-ng.c aircrack-ng-0.6-inject/src/airodump-ng.c
--- aircrack-ng-0.6/src/airodump-ng.c	2006-06-23 20:21:29.000000000 +0200
+++ aircrack-ng-0.6-inject/src/airodump-ng.c	2006-08-23 13:50:48.000000000 +0200
@@ -1726,6 +1726,11 @@
 			return( WEXITSTATUS(status) );
 		return( 1 );		
 	}
+	else if(strcmp(interface,"rtap0")==0)
+	{
+		printf("Rtap interface detected");
+		return(0);
+	}
 	else
 	{
 		if( G.is_wlanng )
