--- w3af-1.6.54/w3af_console.orig   2016-01-17 15:33:49.649413704 +0800
+++ w3af-1.6.54/w3af_console        2016-01-17 15:34:10.156414862 +0800
@@ -9,8 +9,8 @@
  
 
 # Check if I have all needed dependencies
-from w3af.core.controllers.dependency_check.dependency_check import dependency_check
-dependency_check()
+#from w3af.core.controllers.dependency_check.dependency_check import dependency_check
+#dependency_check()
 
 from w3af.core.ui.console.console_ui import ConsoleUI
 from w3af.core.controllers.misc.get_w3af_version import get_w3af_version
--- w3af-1.6.54/w3af_gui.orig       2016-01-17 15:33:56.250414076 +0800
+++ w3af-1.6.54/w3af_gui    2016-01-17 15:34:17.283415265 +0800
@@ -9,8 +9,8 @@
 
 # Perform the GTK UI dependency check, this will verify that the current system
 # has all the modules required to run w3af (including the core dependencies)
-from w3af.core.ui.gui.dependency_check.dependency_check import dependency_check
-dependency_check()
+#from w3af.core.ui.gui.dependency_check.dependency_check import dependency_check
+#dependency_check()
 
 import w3af.core.controllers.output_manager as om
 
