--- kacpimon/makefile
+++ kacpimon/makefile
@@ -8,7 +8,7 @@
 # release build
 #CFLAGS = -O2 -Wall -Wextra -Wundef -Wshadow -Werror
 # hybrid build, optimized, but with debugging symbols (Debian-style)
-CFLAGS = -g -O2 -Wall -Wextra -Wundef -Wshadow -Werror
+CFLAGS += -Wall -Wextra -Wundef -Wshadow
 
 objects := $(sources:.c=.o)
 
--- Makefile
+++ Makefile
@@ -3,8 +3,6 @@
 # update these numbers for new releases
 VERSION = 2.0.3
 
-OPT = -O2
-
 DESTDIR =
 PREFIX = /usr
 
@@ -29,10 +27,10 @@
 MAN8 = acpid.8 acpi_listen.8
 MAN8GZ = $(MAN8:.8=.8.gz)
 
-DOCS = COPYING Changelog README TESTPLAN TODO 
+DOCS = Changelog README TESTPLAN TODO 
 
-CFLAGS = -W -Wall -Werror -Wundef -Wshadow -D_GNU_SOURCE $(OPT) \
-	-fno-strict-aliasing -g $(DEFS)
+CFLAGS += -W -Wall -Wundef -Wshadow -D_GNU_SOURCE \
+	-fno-strict-aliasing $(DEFS)
 DEFS = -DVERSION="\"$(VERSION)\""
 
 all: $(PROGS)
