diff -Nru rcracki_mt_0.6.5.2_src.orig/ChainWalkContext.cpp rcracki_mt_0.6.5.2_src/ChainWalkContext.cpp
--- rcracki_mt_0.6.5.2_src.orig/ChainWalkContext.cpp	2010-11-13 11:46:28.786198146 +0100
+++ rcracki_mt_0.6.5.2_src/ChainWalkContext.cpp	2010-11-13 11:49:11.765027286 +0100
@@ -91,6 +91,8 @@
 			readCharset = true;
 		else if ( ReadLinesFromFile(GetApplicationPath() + "charset.txt", vLine) )
 			readCharset = true;
+		else if ( ReadLinesFromFile("@@SHARE@@/charset.txt", vLine ) )
+		        readCharset = true;
 	#endif
 
 	if ( readCharset )
