diff -ur a/rcracki_mt/ChainWalkContext.cpp b/rcracki_mt/ChainWalkContext.cpp
--- a/rcracki_mt/ChainWalkContext.cpp	2012-10-29 01:40:58.000000000 +0400
+++ b/rcracki_mt/ChainWalkContext.cpp	2019-11-10 16:04:43.452804848 +0300
@@ -151,6 +151,8 @@
 			readCharset = true;
 		else if ( ReadLinesFromFile(GetApplicationPath() + "charset.txt", vLine) )
 			readCharset = true;
+		else if ( ReadLinesFromFile("@@SHARE@@/charset.txt", vLine ) )
+			readCharset = true;
 		else
 		{
 			vLine = internal_charset;
