diff -urN bokken-1.6.orig/ui/radare_core.py bokken-1.6/ui/radare_core.py
--- bokken-1.6.orig/ui/radare_core.py	2012-01-24 13:59:00.000000000 +0800
+++ bokken-1.6/ui/radare_core.py	2014-05-09 15:55:04.936237151 +0800
@@ -117,7 +117,7 @@
         # Init core
         # Returns True/False (check)
         self.core.file_open(file, 0, 0)
-        self.core.bin_load(None)
+        self.core.bin_load(None, 0)
         #self.core.config.set("asm.bytes", "false")
         self.core.cmd0("e scr.interactive=false")
         self.core.cmd0('e asm.lines=false')
