diff -ur cryptsetup-1.1.0.orig/configure.in cryptsetup-1.1.0/configure.in
--- cryptsetup-1.1.0.orig/configure.in	2010-04-23 21:39:29.000000000 +0300
+++ cryptsetup-1.1.0/configure.in	2010-04-23 21:38:18.000000000 +0300
@@ -62,6 +62,11 @@
 
 dnl ==========================================================================
 
+saved_LIBS=$LIBS
+AC_CHECK_LIB([udev], [udev_queue_get_udev_is_active])
+AC_SUBST(UDEV_LIBS, $LIBS)
+LIBS=$saved_LIBS
+
 AC_ARG_ENABLE(shared-library,
 [  --disable-shared-library
                           disable building of shared cryptsetup library],,
diff -ur cryptsetup-1.1.0.orig/lib/Makefile.am cryptsetup-1.1.0/lib/Makefile.am
--- cryptsetup-1.1.0.orig/lib/Makefile.am	2009-12-30 18:29:28.000000000 +0200
+++ cryptsetup-1.1.0/lib/Makefile.am	2010-04-23 21:41:34.000000000 +0300
@@ -33,6 +33,7 @@
 libcryptsetup_la_LIBADD = \
 	@UUID_LIBS@				\
 	@DEVMAPPER_LIBS@			\
+	@UDEV_LIBS@				\
 	@LIBGCRYPT_LIBS@			\
 	../luks/libluks.la
 
