diff -Naur cryptsetup-1.1.0-rc3/configure.in cryptsetup-1.1.0-rc3-areyoufuckingserious/configure.in
--- cryptsetup-1.1.0-rc3/configure.in	2009-11-16 17:39:37.000000000 -0500
+++ cryptsetup-1.1.0-rc3-areyoufuckingserious/configure.in	2009-11-17 19:49:08.000000000 -0500
@@ -93,9 +93,11 @@
 	saved_LIBS2=$LIBS
 	LIBS="$LIBS -static"
 	# Check if it there is still not some missing dependency like static selinux libs
-	AC_SEARCH_LIBS([dm_set_selinux_context], [devmapper],,
-		[AC_MSG_ERROR([Your system requires SElinux libraries for static compilation.])])
-	LIBS=$saved_LIBS2
+	if test "x$enable_selinux" != xno; then
+		AC_SEARCH_LIBS([dm_set_selinux_context], [devmapper],,
+			[AC_MSG_ERROR([Your system requires SElinux libraries for static compilation.])])
+		LIBS=$saved_LIBS2
+	fi
 fi
 
 LIBS=$saved_LIBS
