--- a/setup.py.orig	2020-05-12 16:02:44.412092000 +0800
+++ b/setup.py	2020-05-14 18:54:56.272896810 +0800
@@ -6,17 +6,7 @@
  'gvm.protocols',
  'gvm.protocols.gmpv7',
  'gvm.protocols.gmpv8',
- 'gvm.protocols.gmpv9',
- 'tests',
- 'tests.connections',
- 'tests.protocols',
- 'tests.protocols.gmp',
- 'tests.protocols.gmpv7',
- 'tests.protocols.gmpv8',
- 'tests.protocols.gmpv9',
- 'tests.protocols.osp',
- 'tests.transforms',
- 'tests.utils']
+ 'gvm.protocols.gmpv9']
 
 package_data = \
 {'': ['*']}
