diff -ur a/pdfid.py b/pdfid.py
--- a/pdfid.py	2019-09-30 20:44:40.000000000 +0300
+++ b/pdfid.py	2019-10-03 12:28:14.005831243 +0300
@@ -367,7 +367,7 @@
 def ParseINIFile():
     oConfigParser = ConfigParser.ConfigParser(allow_no_value=True)
     oConfigParser.optionxform = str
-    oConfigParser.read(os.path.join(os.path.dirname(sys.argv[0]), 'pdfid.ini'))
+    oConfigParser.read(os.path.join('/etc/pdfid/pdfid.ini'))
     keywords = []
     if oConfigParser.has_section('keywords'):
         for key, value in oConfigParser.items('keywords'):
