--- ipv6-toolkit-2.0/work/ipv6toolkit-v2.0/tools/scan6.c.orig	2015-04-05 16:03:21.000000000 +0200
+++ ipv6-toolkit-2.0/work/ipv6toolkit-v2.0/tools/scan6.c	2015-04-06 12:31:01.000000000 +0200
@@ -56,6 +56,13 @@
 #include "ipv6toolkit.h"
 #include "libipv6.h"
 
+#ifndef OUI_DATABASE // keep this the same as the Makefiles
+#define OUI_DATABASE "/usr/share/ipv6toolkit/oui.txt"
+#endif
+#ifndef SNPN_DATABASE // keep this the same as the Makefiles
+#define SNPN_DATABASE "/usr/share/ipv6toolkit/service-names-port-numbers.csv"
+#endif
+
 /* #define DEBUG */
 
 /* Function prototypes */
@@ -5565,10 +5572,10 @@ int process_config_file(const char *path
 	fclose(fp);
 
 	if(!fname_f)
-		strncpy(fname, "/usr/local/share/ipv6toolkit/oui.txt", MAX_FILENAME_SIZE-1);
+		strncpy(fname, OUI_DATABASE, MAX_FILENAME_SIZE-1);
 
 	if(!portsfname_f)
-		strncpy(fname, "/usr/local/share/ipv6toolkit/service-names-port-numbers.csv", MAX_FILENAME_SIZE-1);
+		strncpy(fname, SNPN_DATABASE, MAX_FILENAME_SIZE-1);
 
 	return(1);
 }
