From c11a5dd7d343b101c2cceab03c31f2f59adadf70 Mon Sep 17 00:00:00 2001
From: pancake <pancake@nopcode.org>
Date: Thu, 9 Jul 2015 16:43:38 +0200
Subject: [PATCH] Use Ruby gemdir as installation path

---
 Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index 695d8b1..db05a25 100644
--- a/Makefile
+++ b/Makefile
@@ -199,12 +199,12 @@ install-go:
 install-java:
 	cd java && ${MAKE} install
 
+RUBYPATH=$(shell gem environment gemdir | sed -e s,gems/,,)
 install-ruby:
-	for a in 1.8 1.9.1; do \
-		mkdir -p ${DESTDIR}${PREFIX}/lib/ruby/$$a/r2 ; \
-		echo "Installing ruby$$a r2 modules..." ; \
-		cp -rf ruby/* ${DESTDIR}${PREFIX}/lib/ruby/$$a/r2 ; \
-	done
+	echo "Installing radare2 Ruby modules..."
+	rm -rf ${DESTDIR}${RUBYPATH}/r2
+	mkdir -p ${DESTDIR}${RUBYPATH}/r2
+	cp -rf ruby/* ${DESTDIR}${RUBYPATH}/r2
 
 install-perl:
 	# hack for slpm
