Index: ipw2200-1.2.0/Makefile
===================================================================
--- ipw2200-1.2.0.orig/Makefile
+++ ipw2200-1.2.0/Makefile
@@ -153,14 +153,8 @@ clean:
 distclean: clean
 	rm -f tags TAGS
 
-TMP=$(PWD)/tmp
-MODVERDIR=$(TMP)/.tmp_versions
-
 modules:
-	mkdir -p $(MODVERDIR)
-	-cp $(IEEE80211_MODVERDIR)/*.mod $(MODVERDIR)
-	-cp $(IEEE80211_MODVERDIR)/../Modules.symvers $(PWD)
-	$(MAKE) -C $(KSRC) M=$(PWD) MODVERDIR=$(MODVERDIR) modules
+	$(MAKE) -C $(KSRC) M=$(PWD) modules
 
 
 patch_kernel:
