diff -bur Source-Old/WebCore/ChangeLog Source/WebCore/ChangeLog
--- Source-Old/WebCore/ChangeLog	2014-01-03 17:02:12.986019512 -0500
+++ Source/WebCore/ChangeLog	2014-01-03 17:05:19.242746641 -0500
@@ -1,3 +1,19 @@
+2013-12-02  Andres Gomez  <agomez@igalia.com>
+
+        [GTK] Fails to build with freetype 2.5.1
+        https://bugs.webkit.org/show_bug.cgi?id=125074
+
+        Reviewed by Carlos Garcia Campos.
+
+        FreeType specifies a canonical way of including their own
+        headers. Now, we are following this recommendation so the
+        compilation won't be broken again due to an upgrade in FeeType's
+        including paths.
+
+        * platform/graphics/freetype/FontPlatformDataFreeType.cpp:
+        * platform/graphics/freetype/SimpleFontDataFreeType.cpp:
+        * platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp:
+
 2013-05-03  Xabier Rodriguez Calvar  <calvaris@igalia.com>
 
         [GStreamer] GStreamer log crashes in MediaPlayerPrivateGStreamerBase because of uninitialized category
diff -bur Source-Old/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp Source/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp
--- Source-Old/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp	2014-01-03 17:02:12.965019204 -0500
+++ Source/WebCore/platform/graphics/freetype/FontPlatformDataFreeType.cpp	2014-01-03 17:07:05.353299934 -0500
@@ -29,6 +29,8 @@
 #include <cairo-ft.h>
 #include <cairo.h>
 #include <fontconfig/fcfreetype.h>
+#include <ft2build.h>
+#include FT_TRUETYPE_TABLES_H
 #include <wtf/text/WTFString.h>
 
 #if !PLATFORM(EFL)
diff -bur Source-Old/WebCore/platform/graphics/freetype/SimpleFontDataFreeType.cpp Source/WebCore/platform/graphics/freetype/SimpleFontDataFreeType.cpp
--- Source-Old/WebCore/platform/graphics/freetype/SimpleFontDataFreeType.cpp	2014-01-03 17:02:12.965019204 -0500
+++ Source/WebCore/platform/graphics/freetype/SimpleFontDataFreeType.cpp	2014-01-03 17:06:34.064841946 -0500
@@ -43,6 +43,9 @@
 #include <cairo-ft.h>
 #include <cairo.h>
 #include <fontconfig/fcfreetype.h>
+#include <ft2build.h>
+#include FT_TRUETYPE_TABLES_H
+#include FT_TRUETYPE_TAGS_H
 #include <unicode/normlzr.h>
 #include <wtf/MathExtras.h>
 #include <wtf/unicode/Unicode.h>
diff -bur Source-Old/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp Source/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp
--- Source-Old/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp	2014-01-03 17:02:12.963019175 -0500
+++ Source/WebCore/platform/graphics/harfbuzz/HarfBuzzFaceCairo.cpp	2014-01-03 17:10:17.954121668 -0500
@@ -39,8 +39,9 @@
 #include "TextEncoding.h"
 #include <cairo-ft.h>
 #include <cairo.h>
-#include <freetype/freetype.h>
-#include <freetype/tttables.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_TRUETYPE_TABLES_H
 #include <hb.h>
 #include <wtf/text/CString.h>
 
