--- jdk/make/lib/CoreLibraries.gmk.orig	2016-04-23 02:17:13.000000000 +0800
+++ jdk/make/lib/CoreLibraries.gmk	2016-05-21 15:16:57.281302419 +0800
@@ -183,6 +183,12 @@
   endif
 endif
 
+ifeq ($(OPENJDK_TARGET_OS), linux)
+  ifeq ($(OPENJDK_TARGET_CPU), x86_64)
+    BUILD_LIBJAVA_Bits.c_CFLAGS := $(C_O_FLAG_NORM)
+  endif
+endif
+
 $(eval $(call SetupNativeCompilation,BUILD_LIBJAVA, \
     LIBRARY := java, \
     OUTPUT_DIR := $(INSTALL_LIBRARIES_HERE), \
