--- a/src/hexview/flagsdelegate.h	2026-06-07 23:50:22.625458163 +0800
+++ b/src/hexview/flagsdelegate.h	2026-06-07 23:50:22.626194997 +0800
@@ -12,9 +12,9 @@
 
 public:
     QString comment(quint64 offset, quint8 b,
-                    const QHexView* hexview) const override;
+                    const QHexView* hexview) const;
     bool renderByte(quint64 offset, quint8 b, QHexCharFormat& outcf,
-                    const QHexView* hexview) const override;
+                    const QHexView* hexview) const;
 
 private:
     const RDFlagsBuffer* m_flags{nullptr};
