diff --git a/CMakeLists.txt b/CMakeLists.txt
index e19a105..d0cd682 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.25)
 
 project(redasm-gui)
 
+include(GNUInstallDirs)
+
 if(NOT TARGET redasm::redasm)
     find_package(redasm REQUIRED)
 endif()
@@ -86,6 +88,7 @@ target_sources(${PROJECT_NAME}
 target_compile_definitions(${PROJECT_NAME}
     PRIVATE
       SHARED_OBJECT_EXT="${CMAKE_SHARED_LIBRARY_SUFFIX}"
+      $<$<NOT:$<BOOL:${WIN32}>>:REDASM_PLUGIN_DIR="${CMAKE_INSTALL_FULL_LIBDIR}/redasm/plugins">
 )
 
 target_include_directories(${PROJECT_NAME}
diff --git a/src/main.cpp b/src/main.cpp
index d1d640b..bafbcc7 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -56,6 +56,10 @@ void configure_searchpaths() {
         append_unique(utils::search_paths, APP_DIR + "/plugins");
         append_unique(utils::kb_search_paths, (APP_DIR + "/kb").toUtf8());
 
+#if defined(REDASM_PLUGIN_DIR)
+        append_unique(utils::search_paths, QString::fromUtf8(REDASM_PLUGIN_DIR));
+#endif
+
         for(const QString& sp : QStandardPaths::standardLocations(QStandardPaths::AppDataLocation)) {
             append_unique(utils::search_paths, sp + "/plugins");
             append_unique(utils::kb_search_paths, (sp + "/kb").toUtf8());
