--- a/cmake/Compiler.cmake	2026-06-07 23:39:58.301194777 +0800
+++ b/cmake/Compiler.cmake	2026-06-07 23:39:58.302395470 +0800
@@ -1,4 +1,5 @@
-set_target_properties(${PROJECT_NAME}
+function(setup_compiler TARGET)
+set_target_properties(${TARGET}
     PROPERTIES
         EXPORT_COMPILE_COMMANDS ON
         CXX_STANDARD_REQUIRED YES
@@ -6,7 +7,7 @@
         CXX_STANDARD 17
 )
 
-target_compile_features(${PROJECT_NAME}
+target_compile_features(${TARGET}
     PRIVATE
         cxx_std_17
 )
@@ -54,14 +55,15 @@
     )
 endif()
 
-target_compile_options(${PROJECT_NAME}
+target_compile_options(${TARGET}
     PRIVATE
         $<$<CXX_COMPILER_ID:MSVC>:${MSVC_COMPILE_OPTIONS}>
         $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:${NOT_MSVC_COMPILE_OPTIONS}>
 )
 
-target_link_options(${PROJECT_NAME}
+target_link_options(${TARGET}
     PUBLIC
         $<$<CXX_COMPILER_ID:MSVC>:${MSVC_LINK_OPTIONS}>
         $<$<NOT:$<CXX_COMPILER_ID:MSVC>>:${NOT_MSVC_LINK_OPTIONS}>
 )
+endfunction()
