diff -uNr cowpatty-4.6/cowpatty.c cowpatty-4.6-fixup3/cowpatty.c
--- cowpatty-4.6/cowpatty.c	2009-07-03 08:15:50.000000000 -0700
+++ cowpatty-4.6-fixup3/cowpatty.c	2009-07-13 13:56:59.417045457 -0700
@@ -455,17 +455,36 @@
 		return;
 	}
 
-	if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_MD5_RC4) {
-		/* Check for WPA key, and pairwise key type */
-		if (eapolkeyhdr->type != 254 || 
-				(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
-			return;
-		}
-	} else if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_SHA1_AES) {
-		if (eapolkeyhdr->type != 2 ||
-				(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
-			return;
+	if (opt->nonstrict == 0) {
+
+		if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_MD5_RC4) {
+			/* Check for WPA key, and pairwise key type */
+			if (eapolkeyhdr->type != 254 || 
+					(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+				return;
+			}
+		} else if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_SHA1_AES) {
+			if (eapolkeyhdr->type != 2 ||
+					(key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+				return;
+			}
+		}
+
+	} else {
+
+                if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_MD5_RC4) {
+                        /* Check for WPA key, and pairwise key type */
+                        if ((eapolkeyhdr->type != 2 && eapolkeyhdr->type != 254) ||
+                                        (key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+                                return;
+                        }
+                } else if (cdata->ver == WPA_KEY_INFO_TYPE_HMAC_SHA1_AES) {
+                        if ((eapolkeyhdr->type != 2 && eapolkeyhdr->type != 254) ||
+                                        (key_info & WPA_KEY_INFO_KEY_TYPE) == 0) {
+                                return;
+                        }
 		}
+
 	}
 
 	if (opt->nonstrict == 0) {
@@ -480,6 +499,9 @@
 			memcpy(cdata->snonce, eapolkeyhdr->key_nonce,
 			       sizeof(cdata->snonce));
 			cdata->snonceset = 1;
+			memcpy(cdata->replay_counter1,
+			       eapolkeyhdr->replay_counter, 8);
+			cdata->replay_counter1[7] = cdata->replay_counter1[7] + 1;
 
 		/* Check for frame 3 of the 4-way handshake */
 		} else if ((key_info & WPA_KEY_INFO_MIC)
@@ -497,14 +519,16 @@
 			cdata->anonceset = 1;
 			/* We save the replay counter value in the 3rd frame to match
 			   against the 4th frame of the four-way handshake */
-			memcpy(cdata->replay_counter,
+			memcpy(cdata->replay_counter2,
 			       eapolkeyhdr->replay_counter, 8);
 
 		/* Check for frame 4 of the four-way handshake */
 		} else if ((key_info & WPA_KEY_INFO_MIC)
 			  && (key_info & WPA_KEY_INFO_ACK) == 0
 			  && (key_info & WPA_KEY_INFO_INSTALL) == 0
-			  && (memcmp (cdata->replay_counter,
+			  && (memcmp (cdata->replay_counter1,
+			      cdata->replay_counter2, 8) == 0)
+			  && (memcmp (cdata->replay_counter2,
 			      eapolkeyhdr->replay_counter, 8) == 0)) {
 
 			memcpy(cdata->keymic, eapolkeyhdr->key_mic,
@@ -570,6 +594,14 @@
 void dump_all_fields(struct crack_data cdata, struct user_opt *opt)
 {
 
+	printf("replay_counter1 is:");
+	lamont_hdump(cdata.replay_counter1, 8);
+	printf("\n");
+
+	printf("replay_counter2 is:");
+	lamont_hdump(cdata.replay_counter2, 8);
+	printf("\n");
+
 	printf("AA is:");
 	lamont_hdump(cdata.aa, 6);
 	printf("\n");
diff -uNr cowpatty-4.6/cowpatty.h cowpatty-4.6-fixup3/cowpatty.h
--- cowpatty-4.6/cowpatty.h	2009-06-04 06:24:16.000000000 -0700
+++ cowpatty-4.6-fixup3/cowpatty.h	2009-07-13 13:56:02.832241597 -0700
@@ -178,7 +178,8 @@
 	u8 anonceset;
 	u8 keymicset;
 	u8 eapolframeset;
-	u8 replay_counter[8];
+	u8 replay_counter1[8];
+	u8 replay_counter2[8];
 
 	int ver; /* Hashing algo, MD5 or AES-CBC-MAC */
 	int eapolframe_size;
