diff -Naurp winexe-9999/source4/lib/tls/tls.c winexe-9999/source4/lib/tls/tls.c
--- winexe-9999/source4/lib/tls/tls.c	2013-10-21 10:53:59.000000000 +0200
+++ winexe-9999/source4/lib/tls/tls.c	2013-10-21 10:55:45.000000000 +0200
@@ -505,7 +505,9 @@ struct socket_context *tls_init_server(s
 	gnutls_transport_set_ptr(tls->session, (gnutls_transport_ptr)tls);
 	gnutls_transport_set_pull_function(tls->session, (gnutls_pull_func)tls_pull);
 	gnutls_transport_set_push_function(tls->session, (gnutls_push_func)tls_push);
+#if GNUTLS_VERSION_NUMBER < 0x020c00
 	gnutls_transport_set_lowat(tls->session, 0);
+#endif
 
 	tls->plain_chars = plain_chars;
 	if (plain_chars) {
@@ -574,7 +576,9 @@ struct socket_context *tls_init_client(s
 	gnutls_transport_set_ptr(tls->session, (gnutls_transport_ptr)tls);
 	gnutls_transport_set_pull_function(tls->session, (gnutls_pull_func)tls_pull);
 	gnutls_transport_set_push_function(tls->session, (gnutls_push_func)tls_push);
+#if GNUTLS_VERSION_NUMBER < 0x020c00
 	gnutls_transport_set_lowat(tls->session, 0);
+#endif
 	tls->tls_detect = false;
 
 	tls->output_pending  = false;
