diff --git a/qflipper_common.pri b/qflipper_common.pri
index e24d6acd..4a99e09d 100644
--- a/qflipper_common.pri
+++ b/qflipper_common.pri
@@ -25,20 +25,11 @@ unix:!macx {
     error("Unsupported OS or compiler")
 }
 
-GIT_VERSION = $$system("git describe --tags --abbrev=0","lines", HAS_VERSION)
-!equals(HAS_VERSION, 0) {
-    GIT_VERSION = unknown
-}
+GIT_VERSION = 1.1.2
 
-GIT_COMMIT = $$system("git rev-parse --short=8 HEAD","lines", HAS_COMMIT)
-!equals(HAS_COMMIT, 0) {
-    GIT_COMMIT = unknown
-}
+GIT_COMMIT = 78c52175
 
-GIT_TIMESTAMP = $$system("git log -1 --pretty=format:%ct","lines", HAS_TIMESTAMP)
-!equals(HAS_TIMESTAMP, 0) {
-    GIT_TIMESTAMP = 0
-}
+GIT_TIMESTAMP = 1660050636
 
 DEFINES += APP_NAME=\\\"$$NAME\\\" \
            APP_VERSION=\\\"$$GIT_VERSION\\\" \
