diff --git a/cli/cli.pro b/cli/cli.pro
index 6b5c27ac..4949fe70 100644
--- a/cli/cli.pro
+++ b/cli/cli.pro
@@ -11,13 +11,13 @@ CONFIG -= app_bundle

 unix|win32 {
     LIBS += \
-        -L$$OUT_PWD/../3rdparty/ -l3rdparty \
+        -lprotobuf-nanopb \
         -L$$OUT_PWD/../plugins/ -lflipperproto0 \
         -L$$OUT_PWD/../backend/ -lbackend \
         -L$$OUT_PWD/../dfu/ -ldfu

     contains(CONFIG, static): LIBS += \
-        -L$$OUT_PWD/../3rdparty/ -l3rdparty \
+        -lprotobuf-nanopb \
         -L$$OUT_PWD/../plugins/ -lflipperproto0
 }

diff --git a/plugins/flipperproto0/flipperproto0.pro b/plugins/flipperproto0/flipperproto0.pro
index 1de45121..d8eab59b 100644
--- a/plugins/flipperproto0/flipperproto0.pro
+++ b/plugins/flipperproto0/flipperproto0.pro
@@ -12,8 +12,7 @@ CONFIG += plugin c++11

 VERSION = 0.0.0

-INCLUDEPATH += $$PWD/../protobufinterface \
-    $$PWD/../../3rdparty/nanopb
+INCLUDEPATH += $$PWD/../protobufinterface

 HEADERS += \
     guirequest.h \
@@ -59,7 +58,7 @@ SOURCES += \
     systemresponse.cpp

 unix|win32 {
-    LIBS += -L$$OUT_PWD/../../3rdparty/ -l3rdparty
+    LIBS += -lprotobuf-nanopb
 }

 DEFINES += PB_ENABLE_MALLOC
diff --git a/qFlipper.pro b/qFlipper.pro
index 1bb97627..8b2a6c44 100644
--- a/qFlipper.pro
+++ b/qFlipper.pro
@@ -1,7 +1,6 @@
 TEMPLATE = subdirs

 SUBDIRS += \
-    3rdparty \
     application \
     backend \
     dfu \
@@ -11,4 +10,3 @@ SUBDIRS += \
 backend.depends = dfu plugins
 application.depends = backend
 cli.depends = backend
-plugins.depends = 3rdparty
