--- sleuthkit-3.2.3/tsk3/auto/auto_db.cpp
+++ sleuthkit-3.2.3/tsk3/auto/auto_db.cpp
@@ -14,7 +14,7 @@
  */
 
 #include "tsk_auto_i.h"
-#include "sqlite3.h"
+#include <sqlite3.h>
 #include <string.h>
 
 #define TSK_SCHEMA_VER 1
--- sleuthkit-3.2.3/tsk3/auto/Makefile.am
+++ sleuthkit-3.2.3/tsk3/auto/Makefile.am
@@ -3,7 +3,8 @@
 
 noinst_LTLIBRARIES = libtskauto.la
 # Note that the .h files are in the top-level Makefile
-libtskauto_la_SOURCES = auto.cpp tsk_auto_i.h auto_db.cpp sqlite3.c sqlite3.h
+libtskauto_la_SOURCES = auto.cpp tsk_auto_i.h auto_db.cpp
+libtskauto_la_LIBADD = -lsqlite3
 
 indent:
 	indent *.cpp *.h
